<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to metR • metR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">metR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/metR_vignette.html">Introduction to metR</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/geneticsMiNIng/metR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="metR_vignette_files/htmlwidgets-0.9/htmlwidgets.js"></script><script src="metR_vignette_files/jquery-1.11.1/jquery.min.js"></script><script src="metR_vignette_files/datatables-binding-0.1/datatables.js"></script><script src="metR_vignette_files/datatables-1.10.7/jquery.dataTables.min.js"></script><link href="metR_vignette_files/datatables-default-1.10.7/dataTables.extra.css" rel="stylesheet">
<link href="metR_vignette_files/datatables-default-1.10.7/jquery.dataTables.min.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Introduction to metR</h1>
                        <h4 class="author">Aleksandra Brodecka, Przemysław Biecek</h4>
            
            <h4 class="date">2017-11-28</h4>
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>metR package allows comprehensive data analysis from methylation studies. By metR we can find create regions, get basic statistics about them, plot methylation rate or arrange created regions from the most interesing by diffrent methods.</p>
<p>We will use the following packages:</p>
</div>
<div id="small-probes---examples-from-description" class="section level1">
<h1 class="hasAnchor">
<a href="#small-probes---examples-from-description" class="anchor"></a>Small probes - examples from description</h1>
<p>sample.1 and sample.2 are datasets containing a part of the methylation data from some patient’s healthy and diseased tissue. See ?sample.1 or ?sample.2 if you want more details about columns.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"sample.1"</span>)
<span class="kw">data</span>(<span class="st">"sample.2"</span>)</code></pre></div>
<div id="preprocessing" class="section level3">
<h3 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>Preprocessing</h3>
<p>We use preprocessing function to binding these probes on chromosome and position:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/preprocessing.html">preprocessing</a></span>(sample.<span class="dv">1</span>, sample.<span class="dv">2</span>)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data, <span class="dv">10</span>))</code></pre></div>
<div id="htmlwidget-dd8e6b365e18dc10b8f5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-dd8e6b365e18dc10b8f5">{"x":{"data":[["1","2","3","4","5","6","7","8","9","10"],["chr1","chr1","chr1","chr1","chr1","chr1","chr1","chr1","chr1","chr1"],[12779,12779,12807,12807,12867,12867,14889,14889,14948,14948],["x","y","x","y","x","y","x","y","x","y"],[4,7,4,7,2,7,4,3,1,2],[4,6,2,7,2,4,0,0,1,2],[0,1,2,0,0,3,4,3,0,0],[1,0.857142857142857,0.5,1,1,0.571428571428571,0,0,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="creating-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#creating-regions" class="anchor"></a>Creating regions</h3>
<p>Now we can create region. After creation they will be tested if there is significant difference between two probes in methylation rate.</p>
<p>One way is to use create.tiles.fixed.length function, See ?create.tiles.fixed.length if you want some details.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles.<span class="dv">0</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.tiles.fixed.length.html">create.tiles.fixed.length</a></span>(data, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> T)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data.tiles.<span class="dv">0</span>))</code></pre></div>
<div id="htmlwidget-b521e0345b0b4e02c883" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b521e0345b0b4e02c883">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[12779,12779,12807,12807,12867,12867],["x","y","x","y","x","y"],[4,7,4,7,2,7],[4,6,2,7,2,4],[0,1,2,0,0,3],[1,0.857142857142857,0.5,1,1,0.571428571428571],[12,12,12,12,12,12],[13,13,13,13,13,13]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n      <th>tiles<\/th>\n      <th>tiles.common<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles.<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.tiles.fixed.length.html">create.tiles.fixed.length</a></span>(data, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> F)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data.tiles.<span class="dv">1</span>))</code></pre></div>
<div id="htmlwidget-0f21c098e111191c4e21" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-0f21c098e111191c4e21">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[12779,12779,12807,12807,12867,12867],["x","y","x","y","x","y"],[4,7,4,7,2,7],[4,6,2,7,2,4],[0,1,2,0,0,3],[1,0.857142857142857,0.5,1,1,0.571428571428571],[12,12,12,12,12,12]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n      <th>tiles<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><p>We can also create region by create.tiles.max.gap function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles.<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.tiles.max.gap.html">create.tiles.max.gap</a></span>(data, <span class="dt">gaps.length =</span> <span class="dv">100</span>)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data.tiles.<span class="dv">2</span>))</code></pre></div>
<div id="htmlwidget-b6484dc5734f6bebd0c4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b6484dc5734f6bebd0c4">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[12779,12779,12807,12807,12867,12867],["x","y","x","y","x","y"],[4,7,4,7,2,7],[4,6,2,7,2,4],[0,1,2,0,0,3],[1,0.857142857142857,0.5,1,1,0.571428571428571],[1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n      <th>tiles<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="basic-statistics" class="section level3">
<h3 class="hasAnchor">
<a href="#basic-statistics" class="anchor"></a>Basic statistics</h3>
<p>We run get.stats function to get basic statisitcs about created regions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">stats.<span class="dv">0</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get.stats.html">get.stats</a></span>(data.tiles.<span class="dv">0</span>)
stats.<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get.stats.html">get.stats</a></span>(data.tiles.<span class="dv">1</span>)
stats.<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get.stats.html">get.stats</a></span>(data.tiles.<span class="dv">2</span>)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(stats.<span class="dv">2</span>))</code></pre></div>
<div id="htmlwidget-14d25e999743c017efad" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-14d25e999743c017efad">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[12779,14889,15380,15643,51789,61954],[12867,15240,15526,15720,51978,61955],[3,15,10,2,13,2],[1,1,1,0.869565217391304,1,0.892857142857143],[1,1,1,0.818181818181818,1,0.888888888888889],[0.833333333333333,0.766666666666667,0.941304347826087,0.434782608695652,0.923076923076923,0.479761904761905],[0.80952380952381,0.692222222222222,0.985714285714286,0.431818181818182,1,0.444444444444444],[0.5,0,0.75,0,0,0.0666666666666667],[0.571428571428571,0,0.857142857142857,0.0454545454545455,1,0],[0.288675134594813,0.312122788211042,0.104408195097553,0.614875461901346,0.277350098112615,0.584204888266029],[0.218217890235992,0.374105635422706,0.0451753951452626,0.546400694553241,0,0.628539361054709],[0.0238095238095238,0.0744444444444444,0.0444099378881988,0.00296442687747034,0.0769230769230769,0.0353174603174604],[0.350905086743638,0.820348734500687,0.655065785132179,0.0725005471894593,0.82055444696673,0.430545454545455]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>meth.cov<\/th>\n      <th>meth.max_x<\/th>\n      <th>meth.max_y<\/th>\n      <th>meth.mean_x<\/th>\n      <th>meth.mean_y<\/th>\n      <th>meth.min_x<\/th>\n      <th>meth.min_y<\/th>\n      <th>meth.sd_x<\/th>\n      <th>meth.sd_y<\/th>\n      <th>meth.diff<\/th>\n      <th>quantile<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12,13,14]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><p>Which function gives us greater regions?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">c</span>(<span class="kw">median</span>(stats.<span class="dv">0</span><span class="op">$</span>meth.cov),<span class="kw">median</span>(stats.<span class="dv">1</span><span class="op">$</span>meth.cov),<span class="kw">median</span>(stats.<span class="dv">2</span><span class="op">$</span>meth.cov))</code></pre></div>
<pre><code>## [1] 11 10  6</code></pre>
<p>Or greater methylation difference?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">c</span>(<span class="kw">median</span>(stats.<span class="dv">0</span><span class="op">$</span>meth.diff), <span class="kw">median</span>(stats.<span class="dv">1</span><span class="op">$</span>meth.diff) , <span class="kw">median</span>(stats.<span class="dv">2</span><span class="op">$</span>meth.diff))</code></pre></div>
<pre><code>## [1] 0.05452228 0.05452228 0.07086208</code></pre>
<p>We propose tuning parameters: gaps.length, common and tiles.length depending on how long and how many regions we want to obtain.</p>
</div>
<div id="finding-dmr" class="section level3">
<h3 class="hasAnchor">
<a href="#finding-dmr" class="anchor"></a>Finding DMR</h3>
<p>We can sort regions by 6 methods: Wilcoxon, t-Student, Kolmogorov-Smirnow tests or using logistic regression and its modifications. See ?find.DMR if you want some more details.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find.DMR.html">find.DMR</a></span>(data.tiles.<span class="dv">2</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Wilcoxon'</span>, <span class="st">'Ttest'</span>, <span class="st">'KS'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Mixed'</span>, <span class="st">'Reg.Corr.Mixed'</span>))</code></pre></div>
<p>We get list of six data frames with columns: chr - chromosome, start and end which are min and max position of region and p.value which is p.value from tested method. Logistic regression methods return also data frame with beta.coeff columns which is beta coefficient og grouping variable. In each data frame regions are sorted by most interesing to uninteresing groups.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(result)</code></pre></div>
<pre><code>## [1] "Wilcoxon"       "Ttest"          "KS"             "Reg.Log"       
## [5] "Reg.Mixed"      "Reg.Corr.Mixed"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(result<span class="op">$</span>Wilcoxon))</code></pre></div>
<div id="htmlwidget-8664184f22b85f3d5daa" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8664184f22b85f3d5daa">{"x":{"data":[["1","2","3","4","5","6"],["chr22","chr2","chr22","chr13","chr13","chr13"],[29876228,171573098,38610287,58206873,114064726,58207898],[29877221,171574053,38610789,58207781,114065183,58208141],[1.03478559433607e-28,7.18325565443881e-21,4.24423604145915e-19,7.09319958751152e-17,7.85919775305785e-09,1.18781117187236e-07]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>p.value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><p>We can change ordering logistic regressions from p.value of grouping variable to absolute value of beta coefficient. We must establish parameters: p.value.log.reg, p.value.reg.mixed or p.value.reg.corr.mixed. Function find.DMR returns only regions which have smaller p.values than p.value from adequate methods.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result.<span class="dv">2</span> &lt;-<span class="kw"><a href="../reference/find.DMR.html">find.DMR</a></span>(data.tiles.<span class="dv">2</span>, <span class="kw">c</span>(<span class="st">'Reg.Log'</span>, <span class="st">'Reg.Mixed'</span>, <span class="st">'Reg.Corr.Mixed'</span>), <span class="dt">p.value.log.reg =</span> <span class="fl">0.01</span>, <span class="dt">p.value.reg.mixed =</span> <span class="fl">0.02</span>, <span class="dt">p.value.reg.corr.mixed=</span><span class="fl">0.03</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(result.<span class="dv">2</span><span class="op">$</span>Reg.Log))</code></pre></div>
<div id="htmlwidget-7d72c5ca7e4b2dc343d3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7d72c5ca7e4b2dc343d3">{"x":{"data":[["1","2","3","4","5","6"],["chr6","chr13","chr16","chr13","chr18","chr22"],[35286051,114064726,91645,114064452,74583,38618918],[35286264,114065183,91707,114064537,74610,38619092],[2.15700214927777e-15,1.32177936975371e-141,2.55224753203952e-16,6.93785248214398e-54,0.00809350216738474,1.45581091008184e-09],[-4.63225145099212,-3.22214305242334,2.81069129646555,-2.57922574318026,2.18612129620301,-1.97475507724895]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>p.value<\/th>\n      <th>beta.coef<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="plotting-methylation" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-methylation" class="anchor"></a>Plotting methylation</h3>
<p>We use draw.methylation function: we have to give parameters like: data - data.frame from preprocessing function, chr - chromosome name, start and end which are min and max position of plotted region. Other parameters allow us change graphics parameters from ggplot object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw.methylation.html">draw.methylation</a></span>(data, <span class="dt">chr =</span> <span class="st">'chr2'</span>, <span class="dt">start =</span> <span class="dv">171573000</span>, <span class="dt">end =</span> <span class="dv">171574000</span>, <span class="dt">legend.title =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'bottom'</span>, <span class="dt">plot.title =</span> <span class="dv">17</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>,
                 <span class="dt">axis.text.x =</span> <span class="dv">11</span>, <span class="dt">axis.text.y =</span> <span class="dv">11</span>, 
                 <span class="dt">size.x.dot =</span> <span class="dv">7</span>, <span class="dt">size.y.dot =</span> <span class="dv">4</span>, <span class="dt">legend.text =</span> <span class="dv">8</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-16-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw.methylation.html">draw.methylation</a></span>(data, <span class="dt">chr =</span> <span class="st">'chr2'</span>, <span class="dt">start =</span> <span class="dv">171573000</span>, <span class="dt">end =</span> <span class="dv">171574000</span>, <span class="dt">legend.title =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'bottom'</span>, <span class="dt">plot.title =</span> <span class="dv">17</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>,
                 <span class="dt">axis.text.x =</span> <span class="dv">11</span>, <span class="dt">axis.text.y =</span> <span class="dv">11</span>, 
                 <span class="dt">size.x.dot =</span> <span class="dv">7</span>, <span class="dt">size.y.dot =</span> <span class="dv">4</span>, <span class="dt">bind.probes =</span> F, <span class="dt">smooth.methylation =</span> F, <span class="dt">legend.text =</span> <span class="dv">8</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-16-2.png" width="720"></p>
</div>
</div>
<div id="comprehensive-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#comprehensive-analysis" class="anchor"></a>Comprehensive analysis:</h1>
<p>Data for this example are downloaded from site: <a href="http://www.neuroepigenomics.org/methylomedb/download.html" class="uri">http://www.neuroepigenomics.org/methylomedb/download.html</a> We used 4 control samples: Control1 AC, Control2 AC, Control3 AC and Control4 AC and 4 disease samples: SCZ1 AC, SCZ2 AC, SCZ3 AC, SCZ4 AC. After downloading we summed all results on the same position and chromosome separately from control and disease samples.</p>
<p>In this file we created analysis of DMR in chromosome 1. The code about preprocessing files can be found in examples/prep.MethylomeDB.R in this repo. Result of this script are two data.frames ‘control’ and ‘disease’ that are included in metR package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"control"</span>)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(control))</code></pre></div>
<div id="htmlwidget-813411b77db03abc867e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-813411b77db03abc867e">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[73465,73467,74406,74873,75268,75390],[6,6,10,10,10,10],[1,1,5,5,5,5],[5,5,5,5,5,5],[0.166666666666667,0.166666666666667,0.5,0.5,0.5,0.5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"disease"</span>)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(disease))</code></pre></div>
<div id="htmlwidget-329876f814485cec2f6e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-329876f814485cec2f6e">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[4603,4609,4635,4637,4653,4750],[5,5,5,5,5,5],[2,2,2,2,2,2],[3,3,3,3,3,3],[0.4,0.4,0.4,0.4,0.4,0.4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><div id="preprocess-data" class="section level3">
<h3 class="hasAnchor">
<a href="#preprocess-data" class="anchor"></a>Preprocess data:</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/preprocessing.html">preprocessing</a></span>(control, disease)

DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data))</code></pre></div>
<div id="htmlwidget-a52c18c0e5a75030e3a7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a52c18c0e5a75030e3a7">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[81412,81412,81442,81442,81546,81546],["x","y","x","y","x","y"],[35,76,35,76,35,76],[29,66,29,66,29,66],[6,10,6,10,6,10],[0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><p>Check correlation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(prob <span class="op">==</span><span class="st"> 'x'</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">a =</span> <span class="kw">log</span>((meth <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)<span class="op">/</span>(unmeth <span class="op">+</span><span class="st"> </span><span class="dv">1</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(a) -&gt;<span class="st"> </span>a 
data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(prob <span class="op">==</span><span class="st"> 'y'</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">b =</span> <span class="kw">log</span>((meth <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)<span class="op">/</span>(unmeth <span class="op">+</span><span class="st"> </span><span class="dv">1</span>))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(b) -&gt;<span class="st"> </span>b 
<span class="kw">cor</span>(a,b)</code></pre></div>
<pre><code>## [1] 0.8895702</code></pre>
<p>So we see that these probes are highly correlated.</p>
</div>
<div id="creating-regions-1" class="section level3">
<h3 class="hasAnchor">
<a href="#creating-regions-1" class="anchor"></a>Creating regions</h3>
<p>After preprocessing we can create region. After creation they will be tested if there is significant difference between two probes in methylation rate.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles.<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.tiles.fixed.length.html">create.tiles.fixed.length</a></span>(data, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> F)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data.tiles.<span class="dv">1</span>))</code></pre></div>
<div id="htmlwidget-e32d3dc160e08bf15f83" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e32d3dc160e08bf15f83">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[81412,81412,81442,81442,81546,81546],["x","y","x","y","x","y"],[35,76,35,76,35,76],[29,66,29,66,29,66],[6,10,6,10,6,10],[0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579],[81,81,81,81,81,81]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n      <th>tiles<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles.<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.tiles.fixed.length.html">create.tiles.fixed.length</a></span>(data, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> T)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data.tiles.<span class="dv">2</span>))</code></pre></div>
<div id="htmlwidget-1b66f0a3d0441344babd" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1b66f0a3d0441344babd">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[81412,81412,81442,81442,81546,81546],["x","y","x","y","x","y"],[35,76,35,76,35,76],[29,66,29,66,29,66],[6,10,6,10,6,10],[0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579],[81,81,81,81,81,81],[81,81,81,81,82,82]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n      <th>tiles<\/th>\n      <th>tiles.common<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles.<span class="dv">3</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.tiles.max.gap.html">create.tiles.max.gap</a></span>(data, <span class="dt">gaps.length =</span> <span class="dv">100</span>)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(data.tiles.<span class="dv">3</span>))</code></pre></div>
<div id="htmlwidget-e443112e1eb7e71c36d3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e443112e1eb7e71c36d3">{"x":{"data":[["chr1.1","chr1.2","chr1.3","chr1.4","chr1.5","chr1.6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[81412,81412,81442,81442,81546,81546],["x","y","x","y","x","y"],[35,76,35,76,35,76],[29,66,29,66,29,66],[6,10,6,10,6,10],[0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579,0.828571428571429,0.868421052631579],[1,1,1,1,2,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>poz<\/th>\n      <th>prob<\/th>\n      <th>no<\/th>\n      <th>meth<\/th>\n      <th>unmeth<\/th>\n      <th>meth.rate<\/th>\n      <th>tiles<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><p>Now we can compare how perform our functions. We got basic statistics about two probes. This is helpful if we want check coverage of created regions or methylation difference.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">stats.<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get.stats.html">get.stats</a></span>(data.tiles.<span class="dv">1</span>)
stats.<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get.stats.html">get.stats</a></span>(data.tiles.<span class="dv">2</span>)
stats.<span class="dv">3</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get.stats.html">get.stats</a></span>(data.tiles.<span class="dv">3</span>)
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(stats.<span class="dv">3</span>))</code></pre></div>
<div id="htmlwidget-562a8c88abc6e075e92b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-562a8c88abc6e075e92b">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[81412,81546,81698,81984,82254,82603],[81442,81546,81863,82025,82254,82625],[2,1,4,3,1,2],[0.828571428571429,0.828571428571429,0.828571428571429,0.828571428571429,0.823529411764706,0.727272727272727],[0.868421052631579,0.868421052631579,0.868421052631579,0.868421052631579,0.855263157894737,0.884615384615385],[0.828571428571429,0.828571428571429,0.828571428571429,0.825210084033613,0.823529411764706,0.727272727272727],[0.868421052631579,0.868421052631579,0.868421052631579,0.867251461988304,0.855263157894737,0.884615384615385],[0.828571428571429,0.828571428571429,0.828571428571429,0.823529411764706,0.823529411764706,0.727272727272727],[0.868421052631579,0.868421052631579,0.868421052631579,0.866666666666667,0.855263157894737,0.884615384615385],[0,null,0,0.00291100976062002,null,0],[0,null,0,0.00101289520910461,null,0],[0.0398496240601504,0.0398496240601504,0.0398496240601504,0.0420413779546908,0.031733746130031,0.157342657342657],[0.467264890222206,0.40515964448383,0.533362910723889,0.523590313926598,0.340059709139977,0.879514242343834]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>meth.cov<\/th>\n      <th>meth.max_x<\/th>\n      <th>meth.max_y<\/th>\n      <th>meth.mean_x<\/th>\n      <th>meth.mean_y<\/th>\n      <th>meth.min_x<\/th>\n      <th>meth.min_y<\/th>\n      <th>meth.sd_x<\/th>\n      <th>meth.sd_y<\/th>\n      <th>meth.diff<\/th>\n      <th>quantile<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10,11,12,13,14]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">stats.all &lt;-<span class="st"> </span>gdata<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/gdata/topics/combine">combine</a></span>(stats.<span class="dv">1</span>, stats.<span class="dv">2</span>, stats.<span class="dv">3</span>)

<span class="kw">ggplot</span>(stats.all, <span class="kw">aes</span>(<span class="dt">x =</span> meth.diff, <span class="dt">group =</span> source, <span class="dt">col =</span> source)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">'Distribution of methylation difference'</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(
      <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">15</span>),
      <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">13</span>),
      <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">13</span>),
      <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">11</span>),
      <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">11</span>),
      <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>),
      <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>))</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-22-2.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(stats.all, <span class="kw">aes</span>(<span class="dt">x =</span> meth.cov, <span class="dt">group =</span> source, <span class="dt">col =</span> source)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>,<span class="dv">100</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">'Distribution of coverage regions'</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(
      <span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">15</span>),
      <span class="dt">axis.title.x =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">13</span>),
      <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">13</span>),
      <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">11</span>),
      <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">11</span>),
      <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>),
      <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>))</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-22-3.png" width="576"></p>
<p>We also can join e.g stats.3 and data.tiles.3 on chromosome, start and end column and analyzing regions only if they coverage is greater than some specyfic values or other condition.</p>
</div>
<div id="finding-dmr-1" class="section level3">
<h3 class="hasAnchor">
<a href="#finding-dmr-1" class="anchor"></a>Finding DMR</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># eval = F</span>
result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find.DMR.html">find.DMR</a></span>(data.tiles.<span class="dv">3</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Wilcoxon'</span>, <span class="st">'Ttest'</span>, <span class="st">'KS'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Mixed'</span>, <span class="st">'Reg.Corr.Mixed'</span>))</code></pre></div>
<p>Results of running function above are included in package. So now we only load data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">'find.DMR.results'</span>)
<span class="kw">names</span>(find.DMR.results)</code></pre></div>
<pre><code>## [1] "Wilcoxon"       "Ttest"          "KS"             "Reg.Log"       
## [5] "Reg.Mixed"      "Reg.Corr.Mixed"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(find.DMR.results<span class="op">$</span>Wilcoxon, <span class="dv">10</span>))</code></pre></div>
<div id="htmlwidget-e2536366314711a3adc8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e2536366314711a3adc8">{"x":{"data":[["1","2","3","4","5","6","7","8","9","10"],["chr1","chr1","chr1","chr1","chr1","chr1","chr1","chr1","chr1","chr1"],[4613764,923898,243382926,1210988,7685924,1899539,1232466,887617,973037,27746337],[4616966,927566,243387905,1215473,7688665,1903783,1234619,892586,977565,27751275],[1.00518496946393e-72,1.54545057164442e-71,3.68966634509675e-64,1.18972432894834e-61,6.07549247518537e-56,1.18056047191944e-53,2.23981972534708e-51,3.4805769051566e-50,2.6330385315186e-49,1.18935774846489e-48]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>p.value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="plotting-methylation-1" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-methylation-1" class="anchor"></a>Plotting methylation</h3>
<p>Based on draw.methylation function we plotted ten best result from find.DMR.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">top.n &lt;-<span class="st"> </span><span class="cf">function</span>(find.DMR.results, data, title){
  p &lt;-<span class="st"> </span><span class="kw">list</span>()
  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>){
  p[[i]] &lt;-<span class="st"> </span><span class="kw"><a href="../reference/draw.methylation.html">draw.methylation</a></span>(data, <span class="dt">chromosom =</span> <span class="st">'chr1'</span>, <span class="dt">start =</span> <span class="kw">pull</span>(find.DMR.results[i, <span class="st">'start'</span>]), <span class="dt">end =</span> <span class="kw">pull</span>(find.DMR.results[i, <span class="st">'end'</span>]), <span class="dt">bind.probes =</span> F,
        <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'none'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>)
  }
  
  <span class="kw">grid.arrange</span>(<span class="kw">arrangeGrob</span>(p[[<span class="dv">1</span>]], p[[<span class="dv">2</span>]], <span class="dt">ncol =</span> <span class="dv">2</span>),
  <span class="kw">arrangeGrob</span>(p[[<span class="dv">3</span>]], p[[<span class="dv">4</span>]], <span class="dt">ncol =</span> <span class="dv">2</span>), 
  <span class="kw">arrangeGrob</span>(p[[<span class="dv">5</span>]], p[[<span class="dv">6</span>]], <span class="dt">ncol =</span> <span class="dv">2</span>), 
  <span class="kw">arrangeGrob</span>(p[[<span class="dv">7</span>]], p[[<span class="dv">8</span>]], <span class="dt">ncol =</span> <span class="dv">2</span>),
  <span class="kw">arrangeGrob</span>(p[[<span class="dv">9</span>]], p[[<span class="dv">10</span>]], <span class="dt">ncol =</span> <span class="dv">2</span>),
             <span class="dt">nrow =</span> <span class="dv">5</span>, <span class="dt">top=</span><span class="kw">textGrob</span>(title, <span class="dt">gp=</span><span class="kw">gpar</span>(<span class="dt">fontsize=</span><span class="dv">14</span>,<span class="dt">font=</span><span class="dv">8</span>)))              
}</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Wilcoxon, data, <span class="st">'Top results by Wilcoxon test'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-26-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Ttest, data, <span class="st">'Top results by t-test'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-26-2.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>KS, data, <span class="st">'Top results by K-S test'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-26-3.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Reg.Log, data, <span class="st">'Top results by logistic regression'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-26-4.png" width="720"></p>
<p>We can also sorting regions from logistic regression by effect.size:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results<span class="op">$</span>Reg.Log.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Log <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))</code></pre></div>
<p>The outcome above is equivalent if we run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find.DMR.html">find.DMR</a></span>(data.tiles.<span class="dv">3</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Reg.Log'</span>), <span class="dt">p.value.log.reg =</span> <span class="fl">0.001</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Reg.Log.Beta, data, <span class="st">'Top results by logistic regression'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-29-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Reg.Mixed, data, <span class="st">'Top results by log.reg with random effects'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-30-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results<span class="op">$</span>Reg.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Mixed <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(find.DMR.results<span class="op">$</span>Reg.Mixed.Beta))</code></pre></div>
<div id="htmlwidget-62808cfcaf8e30cea951" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-62808cfcaf8e30cea951">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[144383127,95907546,183043690,166805591,31982418,244107599],[144383300,95908321,183044038,166805855,31982568,244107921],[0,1.07524377536038e-13,1.99380268978731e-07,2.6270387912074e-07,6.47251466323711e-07,8.85786663407691e-07],[-251310.978382319,-5.28758042943697,-5.23397587030184,-5.17725469063811,-5.00962310885061,-4.96057813805511]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>p.value<\/th>\n      <th>beta.coef<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><p>The outcome above is equivalent if we run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find.DMR.html">find.DMR</a></span>(data.tiles.<span class="dv">3</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Reg.Corr.Mixed'</span>), <span class="dt">p.value.log.reg =</span> <span class="fl">0.001</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Reg.Mixed.Beta, data, <span class="st">'Top results by log.reg with random effects sorting on beta'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-33-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Reg.Corr.Mixed, data, <span class="st">'Top results by log.reg with random effects with given corr. matrix'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-34-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results<span class="op">$</span>Reg.Corr.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Corr.Mixed <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))
DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(find.DMR.results<span class="op">$</span>Reg.Corr.Mixed.Beta))</code></pre></div>
<div id="htmlwidget-271450d6f45eb9bc3dc3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-271450d6f45eb9bc3dc3">{"x":{"data":[["1","2","3","4","5","6"],["chr1","chr1","chr1","chr1","chr1","chr1"],[148524292,233212445,15543848,192973259,226173075,45139251],[148524523,233212568,15543909,192973443,226173121,45139518],[5.96722530013537e-12,1.86705982171095e-07,0.000312884100230568,1.14370322309966e-05,9.54232837173343e-05,1.21927438015874e-05],[29.9268546176321,-29.8563766916353,29.2696445665945,28.8664304980949,28.8519419593746,27.4367975371258]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>p.value<\/th>\n      <th>beta.coef<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"callback":null,"filter":"none"},"evals":[],"jsHooks":[]}</script><p>The outcome above is equivalent if we run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find.DMR.html">find.DMR</a></span>(data.tiles.<span class="dv">3</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Reg.Corr.Mixed'</span>), <span class="dt">p.value.log.reg =</span> <span class="fl">0.001</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">top.n</span>(find.DMR.results<span class="op">$</span>Reg.Corr.Mixed.Beta, data, <span class="st">'Top results by log.reg with random effects with given corr. matrix sorting on beta'</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-36-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get_top &lt;-<span class="st"> </span><span class="cf">function</span>(data, n, stats){
  data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span>n) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">left_join</span>(stats, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">'chr'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/select.html">select</a></span>(quantile, meth.diff, meth.cov)
  }

top &lt;-<span class="st"> </span><span class="kw">do.call</span>(gdata<span class="op">::</span>combine,<span class="kw">lapply</span>(find.DMR.results, get_top, <span class="dt">n =</span> <span class="dv">100</span>, <span class="dt">stats =</span> stats.<span class="dv">3</span>))
<span class="kw">ggplot</span>(top, <span class="kw">aes</span>(<span class="dt">x =</span> source, <span class="dt">y =</span> quantile)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> source))  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">'Distribution of rank rate by methods'</span>)  <span class="op">+</span>
<span class="st">   </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">"Set1"</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>()  <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>), <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>),
                           <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>), <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">angle =</span> <span class="dv">20</span>) ,  <span class="dt">legend.position =</span> <span class="st">"None"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">"Method"</span>,<span class="dt">y=</span><span class="st">"Rank rate"</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="st">'Reg.Mixed'</span>,<span class="st">'Reg.Corr.Mixed.Beta'</span>,<span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'KS'</span>, <span class="st">"Wilcoxon"</span>, <span class="st">'Reg.Log.Beta'</span>, <span class="st">'Reg.Mixed.Beta'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Ttest'</span>))</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-37-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(top, <span class="kw">aes</span>(<span class="dt">x =</span> source, <span class="dt">y =</span> meth.diff)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> source))  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">'Distribution of methylation absolute difference by methods'</span>)  <span class="op">+</span>
<span class="st">   </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">"Set1"</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>()  <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>), <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>),
                           <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>), <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">angle =</span> <span class="dv">11</span>) ,  <span class="dt">legend.position =</span> <span class="st">"None"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">"Method"</span>,<span class="dt">y=</span><span class="st">"Methylation difference"</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="st">'Reg.Mixed'</span>,<span class="st">'Reg.Corr.Mixed.Beta'</span>,<span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'KS'</span>, <span class="st">"Wilcoxon"</span>, <span class="st">'Ttest'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Log.Beta'</span>, <span class="st">'Reg.Mixed.Beta'</span>))</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-38-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(top, <span class="kw">aes</span>(<span class="dt">x =</span> source, <span class="dt">y =</span> meth.cov)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> source))  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">'Coverage distribution by methods'</span>)  <span class="op">+</span>
<span class="st">   </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">"Set1"</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>()  <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>), <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>),
                           <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>), <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">angle =</span> <span class="dv">20</span>) ,  <span class="dt">legend.position =</span> <span class="st">"None"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">"Method"</span>,<span class="dt">y=</span><span class="st">"Coverage"</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="st">'Reg.Corr.Mixed.Beta'</span>,<span class="st">'Reg.Log.Beta'</span>,<span class="st">'Reg.Mixed.Beta'</span>, <span class="st">'Reg.Mixed'</span>, <span class="st">"KS"</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'Ttest'</span>, <span class="st">'Wilcoxon'</span>))</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-39-1.png" width="720"></p>
<p>We can show, that these method given very different results. The rank rate is very good if we use Ttest, Reg.Log, Reg.Mixed.Beta and Reg.Log.Beta methods. The biggest differences we see for methylation coverage. Wilcoxon test recommended regions that have the most observations ~ 200. Reg.Corr.Mixed.Beta and Reg.Log.Beta proposed smaller group - about 10 observations. If we check methylation difference, Reg.Log.Beta and Reg.Mixed.Beta present very large difference in recommended regions</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#small-probes---examples-from-description">Small probes - examples from description</a></li>
      <li><a href="#comprehensive-analysis">Comprehensive analysis:</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aleksandra Brodecka, Przemyslaw Biecek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
