<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>metR: a deeper study • metR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top navbar-mi2" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-mi2logo" href="http://mi2.mini.pw.edu.pl/">
        <img src="https://github.com/mi2-warsaw/MI2template/blob/master/inst/pkgdown/assets/MI2logo.jpg?raw=true" alt="MI2" height="46" title="MI2"></a>
      <a class="navbar-brand navbar-mi2" href="../index.html">metR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="navbar-mi2 nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/metR_comparison.html">metR: a deeper study</a>
    </li>
    <li>
      <a href="../articles/metR_step_by_step.html">metR: step by step</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/geneticsMiNIng/metR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>metR: a deeper study</h1>
                        <h4 class="author">Aleksandra Brodecka</h4>
            
            <h4 class="date">2018-01-09</h4>
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <code>metR</code> package contains tools for comprehensive analysis of data from methylation studies. With the <code>metR</code> one can define regions, get basic statistics about methylation in them, plot methylation rate or rank these regions from the most interesting to the least one.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"geneticsMiNIng/metR"</span>)
<span class="kw">library</span>(metR)
<span class="kw">library</span>(kableExtra)
<span class="kw">library</span>(knitr)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(metR)</code></pre></div>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Loading required package: dplyr</code></pre>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre><code>## Loading required package: lme4</code></pre>
<pre><code>## Loading required package: Matrix</code></pre>
<pre><code>## Loading required package: nlme</code></pre>
<pre><code>## 
## Attaching package: 'nlme'</code></pre>
<pre><code>## The following object is masked from 'package:lme4':
## 
##     lmList</code></pre>
<pre><code>## The following object is masked from 'package:dplyr':
## 
##     collapse</code></pre>
<pre><code>## Loading required package: magrittr</code></pre>
<pre><code>## Loading required package: tidyr</code></pre>
<pre><code>## 
## Attaching package: 'tidyr'</code></pre>
<pre><code>## The following object is masked from 'package:magrittr':
## 
##     extract</code></pre>
<pre><code>## The following object is masked from 'package:Matrix':
## 
##     expand</code></pre>
<pre><code>## Loading required package: MASS</code></pre>
<pre><code>## 
## Attaching package: 'MASS'</code></pre>
<pre><code>## The following object is masked from 'package:dplyr':
## 
##     select</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(kableExtra)
<span class="kw">library</span>(knitr)</code></pre></div>
</div>
<div id="the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#the-data" class="anchor"></a>The data</h1>
<p>Data for this example are downloaded from the site: <a href="http://www.neuroepigenomics.org/methylomedb/download.html" class="uri">http://www.neuroepigenomics.org/methylomedb/download.html</a>. We used 4 control samples: Control1 AC, Control2 AC, Control3 AC and Control4 AC and 4 disease samples: SCZ1 AC, SCZ2 AC, SCZ3 AC, SCZ4 AC.</p>
<p>In the <code>metR</code> package these samples are available in the <code>schizophrenia</code> data frame. Each row is a single methylation probe. Data from all control/disease samples are combined together.</p>
<p>To speed up calculations and shrink down the size of data we are using only probes from chromosome 1. The R code needed to download data from all samples can be found in the script <a href="https://github.com/geneticsMiNIng/metR/blob/master/examples/prep.MethylomeDB.R" class="uri">https://github.com/geneticsMiNIng/metR/blob/master/examples/prep.MethylomeDB.R</a>. The result of this script is a single data.frame <code>schizophrenia</code> that is included in metR package. The last column <code>category</code> indicates if data are from control samples (‘control’) or from disease samples (‘disease’).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">'schizophrenia'</span>)

<span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(schizophrenia, <span class="dv">4</span>), <span class="st">"html"</span>, <span class="dt">bootstrap_options =</span> <span class="st">"striped"</span> , <span class="dt">full_width =</span> T), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">11</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="dt">color =</span> <span class="st">"black"</span>)</code></pre></div>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
<th style="text-align:left;">
category
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73465
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73467
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
74406
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
74873
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
</tbody>
</table>
</div>
<div id="process-data" class="section level1">
<h1 class="hasAnchor">
<a href="#process-data" class="anchor"></a>Process data</h1>
<p>To obtain interesting regions in subsection <code>Examples of differentially methylated regions</code> we runned 4 functions from <code>metR</code> package:</p>
<ul>
<li><p><code><a href="../reference/preprocessing.html">preprocessing()</a></code> - preprocessing schizophrenia data</p></li>
<li><p><code><a href="../reference/create_tiles_max_gap.html">create_tiles_max_gap()</a></code> - creating regions</p></li>
<li><p><code><a href="../reference/get_stats.html">get_stats()</a></code> - getting basin statistics about created regions</p></li>
<li><p><code><a href="../reference/find_DMR.html">find_DMR()</a></code> - ranking regions from the most interesting to the least one by different methods</p></li>
</ul>
<p>Functions above are described in <code>metR_step_by_step</code> vignettes or in package documentation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">control &lt;-<span class="st"> </span>schizophrenia <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(category <span class="op">==</span><span class="st"> 'control'</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>category)

disease &lt;-<span class="st"> </span>schizophrenia <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(category <span class="op">==</span><span class="st"> 'disease'</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>category)

processed_schizophrenia &lt;-<span class="st"> </span><span class="kw"><a href="../reference/preprocessing.html">preprocessing</a></span>(control, disease)

tiles_schizophrenia &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_tiles_max_gap.html">create_tiles_max_gap</a></span>(processed_schizophrenia, <span class="dt">gaps.length =</span> <span class="dv">100</span>)

stats &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_stats.html">get_stats</a></span>(tiles_schizophrenia)
<span class="kw">data</span>(<span class="st">'find.DMR.results'</span>)
find.DMR.results<span class="op">$</span>Reg.Log.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Log <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))
find.DMR.results<span class="op">$</span>Reg.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Mixed <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))
find.DMR.results<span class="op">$</span>Reg.Corr.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Corr.Mixed <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))

top &lt;-<span class="st"> </span><span class="kw">do.call</span>(gdata<span class="op">::</span>combine,<span class="kw">lapply</span>(find.DMR.results, metR<span class="op">::</span>get_top, <span class="dt">n =</span> <span class="dv">100</span>, <span class="dt">stats =</span> stats))</code></pre></div>
</div>
<div id="examples-of-differentially-methylated-regions" class="section level1">
<h1 class="hasAnchor">
<a href="#examples-of-differentially-methylated-regions" class="anchor"></a>Examples of differentially methylated regions</h1>
<p>Below we are presenting more advance example of comparison of results from different tests. We are showing 8 regions that are results of running metR package on schizophrenia data.</p>
<div id="region-with-maximum-coverage" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-coverage" class="anchor"></a>1. Region with maximum coverage</h3>
<p>This region has the largest coverage from interesting areas. We have 394 observations. This region hasn’t huge methylation diff in two samples but this difference is stable along the entire length.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.max</span>(top<span class="op">$</span>meth.cov),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-6-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
923898
</td>
<td style="text-align:right;color: black;">
927566
</td>
<td style="text-align:right;color: black;">
394
</td>
<td style="text-align:right;color: black;">
0.9
</td>
<td style="text-align:right;color: black;">
0.86
</td>
<td style="text-align:right;color: black;">
0.13
</td>
<td style="text-align:right;color: black;">
0.01
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0.12
</td>
<td style="text-align:right;color: black;">
0.94
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-mean-of-methylation-rate-in-control-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-mean-of-methylation-rate-in-control-sample" class="anchor"></a>2. Region with maximum mean of methylation rate in control sample</h3>
<p>This region has the largest mean of methylation rate in control sample from interesting areas. We have 104 observations and mean of methylation rate is equal 1. This means that all of 104 observations have methylation rate on the level 1!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.max</span>(top<span class="op">$</span>meth.mean_x),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-7-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
3086668
</td>
<td style="text-align:right;color: black;">
3088334
</td>
<td style="text-align:right;color: black;">
104
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.9
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.79
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.76
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.03
</td>
<td style="text-align:right;color: black;">
0.21
</td>
<td style="text-align:right;color: black;">
0.98
</td>
<td style="text-align:left;color: black;">
Ttest
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-minimum-mean-of-methylation-rate-in-disease-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-minimum-mean-of-methylation-rate-in-disease-sample" class="anchor"></a>3. Region with minimum mean of methylation rate in disease sample</h3>
<p>This region has the smallest mean of methylation rate in disease sample from interesting areas. We have 210 observations and mean of methylation rate is equal 0. This means that all of 210 observations have methylation rate on the level 0!</p>
<p>This is reverse example to the previous one.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.min</span>(top<span class="op">$</span>meth.mean_y),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-8-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
20985025
</td>
<td style="text-align:right;color: black;">
20986706
</td>
<td style="text-align:right;color: black;">
210
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.05
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.01
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0.85
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-standard-deviation-of-methylation-rate-in-control-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-standard-deviation-of-methylation-rate-in-control-sample" class="anchor"></a>4. Region with maximum standard deviation of methylation rate in control sample</h3>
<p>We see very interesting regions below. In this example methylation rate start from 0 in both sample and increase to 1 in control sample and 0.75 in disease sample.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.max</span>(top<span class="op">$</span>meth.sd_x),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-9-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
109593468
</td>
<td style="text-align:right;color: black;">
109596927
</td>
<td style="text-align:right;color: black;">
223
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.83
</td>
<td style="text-align:right;color: black;">
0.42
</td>
<td style="text-align:right;color: black;">
0.36
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.47
</td>
<td style="text-align:right;color: black;">
0.39
</td>
<td style="text-align:right;color: black;">
0.07
</td>
<td style="text-align:right;color: black;">
0.87
</td>
<td style="text-align:left;color: black;">
Reg.Log
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-methylation-difference-in-two-probes" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-methylation-difference-in-two-probes" class="anchor"></a>5. Region with maximum methylation difference in two probes</h3>
<p>This area has only two observations but methylation difference between them is huge and equal 0.7.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.max</span>(top<span class="op">$</span>meth.diff),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-10-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
188542753
</td>
<td style="text-align:right;color: black;">
188542830
</td>
<td style="text-align:right;color: black;">
2
</td>
<td style="text-align:right;color: black;">
0.82
</td>
<td style="text-align:right;color: black;">
0.11
</td>
<td style="text-align:right;color: black;">
0.8
</td>
<td style="text-align:right;color: black;">
0.11
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.78
</td>
<td style="text-align:right;color: black;">
0.1
</td>
<td style="text-align:right;color: black;">
0.03
</td>
<td style="text-align:right;color: black;">
0.01
</td>
<td style="text-align:right;color: black;">
0.7
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:left;color: black;">
Reg.Mixed.Beta
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-rank-function" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-rank-function" class="anchor"></a>6. Region with maximum rank function</h3>
<p>This area has maximum rank function. Rank function is based on quantile regression which uses information about number of observations in region. In this example methylation difference between probes is the biggest with respect to number of observations. Methylation difference is close to methylation difference in previous example but has more observation (11 to 2).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.max</span>(top<span class="op">$</span>quantile),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-11-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
79270449
</td>
<td style="text-align:right;color: black;">
79270840
</td>
<td style="text-align:right;color: black;">
11
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.31
</td>
<td style="text-align:right;color: black;">
0.98
</td>
<td style="text-align:right;color: black;">
0.31
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.93
</td>
<td style="text-align:right;color: black;">
0.31
</td>
<td style="text-align:right;color: black;">
0.03
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.66
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:left;color: black;">
Reg.Log.Beta
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-methylation-range-in-control-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-methylation-range-in-control-sample" class="anchor"></a>7. Region with maximum methylation range in control sample</h3>
<p>This region has the biggest methylation range in control sample from interesting areas. We see that methylation rate in control sample starts from 1 and decreases to 0. In disease sample methylation rate starts from level 0.9 and decreases to 0.15. Curious observations is occurence that at the beginning of region we see bigger methylation in control sample and at the end of region reverse case.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.max</span>(top<span class="op">$</span>meth.max_x <span class="op">-</span><span class="st"> </span>top<span class="op">$</span>meth.min_x),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-12-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
62673335
</td>
<td style="text-align:right;color: black;">
62676253
</td>
<td style="text-align:right;color: black;">
233
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.93
</td>
<td style="text-align:right;color: black;">
0.13
</td>
<td style="text-align:right;color: black;">
0.25
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.14
</td>
<td style="text-align:right;color: black;">
0.29
</td>
<td style="text-align:right;color: black;">
0.22
</td>
<td style="text-align:right;color: black;">
0.12
</td>
<td style="text-align:right;color: black;">
0.93
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-methylation-range-in-disease-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-methylation-range-in-disease-sample" class="anchor"></a>8. Region with maximum methylation range in disease sample</h3>
<p>This region has the biggest methylation range in disease sample from interesting areas. In the middle of region we can notice that methylation rate in disease sample is equal to 0 and in control is around 0.25. At the beginning and at the end of regions methylation increases in two probes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">draw_show_table</span>(<span class="kw">which.max</span>(top<span class="op">$</span>meth.max_y <span class="op">-</span><span class="st"> </span>top<span class="op">$</span>meth.min_y),top,processed_schizophrenia)</code></pre></div>
<img src="metR_comparison_files/figure-html/unnamed-chunk-13-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
39342909
</td>
<td style="text-align:right;color: black;">
39345135
</td>
<td style="text-align:right;color: black;">
177
</td>
<td style="text-align:right;color: black;">
0.94
</td>
<td style="text-align:right;color: black;">
0.95
</td>
<td style="text-align:right;color: black;">
0.3
</td>
<td style="text-align:right;color: black;">
0.06
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.24
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.12
</td>
<td style="text-align:right;color: black;">
0.19
</td>
<td style="text-align:right;color: black;">
0.24
</td>
<td style="text-align:right;color: black;">
0.99
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
</div>
<div id="comparing-all-methods" class="section level1">
<h1 class="hasAnchor">
<a href="#comparing-all-methods" class="anchor"></a>Comparing all methods</h1>
<p>Methods above are described in metR_step_by_step or in documentation <code><a href="../reference/find_DMR.html">find_DMR()</a></code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(top, <span class="kw">aes</span>(<span class="dt">x =</span> source, <span class="dt">y =</span> quantile)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> source))  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">'Distribution of rank rate by methods'</span>)  <span class="op">+</span>
<span class="st">   </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">"Set1"</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>()  <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>), <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>),
                           <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>), <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">angle =</span> <span class="dv">20</span>) ,  <span class="dt">legend.position =</span> <span class="st">"None"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">"Method"</span>,<span class="dt">y=</span><span class="st">"Rank rate"</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="st">'Reg.Mixed'</span>,<span class="st">'Reg.Corr.Mixed.Beta'</span>,<span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'KS'</span>, <span class="st">"Wilcoxon"</span>, <span class="st">'Reg.Log.Beta'</span>, <span class="st">'Reg.Mixed.Beta'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Ttest'</span>))</code></pre></div>
<p><img src="metR_comparison_files/figure-html/unnamed-chunk-14-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(top, <span class="kw">aes</span>(<span class="dt">x =</span> source, <span class="dt">y =</span> meth.diff)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> source))  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">'Distribution of methylation absolute difference by methods'</span>)  <span class="op">+</span>
<span class="st">   </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">"Set1"</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>()  <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>), <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>),
                           <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>), <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">angle =</span> <span class="dv">11</span>) ,  <span class="dt">legend.position =</span> <span class="st">"None"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">"Method"</span>,<span class="dt">y=</span><span class="st">"Methylation difference"</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="st">'Reg.Mixed'</span>,<span class="st">'Reg.Corr.Mixed.Beta'</span>,<span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'KS'</span>, <span class="st">"Wilcoxon"</span>, <span class="st">'Ttest'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Log.Beta'</span>, <span class="st">'Reg.Mixed.Beta'</span>))</code></pre></div>
<p><img src="metR_comparison_files/figure-html/unnamed-chunk-15-1.png" width="720"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(top, <span class="kw">aes</span>(<span class="dt">x =</span> source, <span class="dt">y =</span> meth.cov)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> source))  <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="st">'Coverage distribution by methods'</span>)  <span class="op">+</span>
<span class="st">   </span><span class="kw">scale_fill_brewer</span>(<span class="dt">palette =</span> <span class="st">"Set1"</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_minimal</span>()  <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">14</span>), <span class="dt">axis.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">13</span>),
                           <span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>), <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">11</span>, <span class="dt">angle =</span> <span class="dv">20</span>) ,  <span class="dt">legend.position =</span> <span class="st">"None"</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="st">"Method"</span>,<span class="dt">y=</span><span class="st">"Coverage"</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="st">'Reg.Corr.Mixed.Beta'</span>,<span class="st">'Reg.Log.Beta'</span>,<span class="st">'Reg.Mixed.Beta'</span>, <span class="st">'Reg.Mixed'</span>, <span class="st">"KS"</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'Ttest'</span>, <span class="st">'Wilcoxon'</span>))</code></pre></div>
<p><img src="metR_comparison_files/figure-html/unnamed-chunk-16-1.png" width="720"></p>
<p>We can show, that these method given very different results. The rank rate is very good if we use Ttest, Reg.Log, Reg.Mixed.Beta and Reg.Log.Beta methods. The biggest differences we see for methylation coverage. Wilcoxon test recommended regions that have the most observations ~ 200. Reg.Corr.Mixed.Beta and Reg.Log.Beta proposed smaller group - about 10 observations. If we check methylation difference, Reg.Log.Beta and Reg.Mixed.Beta present very large difference in recommended regions.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/session_info">session_info</a></span>()</code></pre></div>
<pre><code>## Session info -------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.4.3 (2017-11-30)
##  system   x86_64, darwin15.6.0        
##  ui       X11                         
<<<<<<< HEAD
##  language pl_PL                       
##  collate  pl_PL.UTF-8                 
##  tz       Poland                      
##  date     2018-01-09</code></pre>
<pre><code>## Packages -----------------------------------------------------------------</code></pre>
<pre><code>##  package      * version date       source         
##  assertthat     0.2.0   2017-04-11 cran (@0.2.0)  
##  backports      1.1.1   2017-09-25 cran (@1.1.1)  
##  base         * 3.4.3   2017-12-01 local          
##  bindr          0.1     2016-11-13 cran (@0.1)    
##  bindrcpp     * 0.2     2017-06-17 cran (@0.2)    
##  colorspace     1.3-2   2016-12-14 cran (@1.3-2)  
##  compiler       3.4.3   2017-12-01 local          
##  datasets     * 3.4.3   2017-12-01 local          
##  devtools       1.13.3  2017-08-02 CRAN (R 3.2.3) 
##  digest         0.6.13  2017-12-14 cran (@0.6.13) 
##  dplyr        * 0.7.4   2017-09-28 cran (@0.7.4)  
##  evaluate       0.10.1  2017-06-24 cran (@0.10.1) 
##  gdata          2.18.0  2017-06-06 CRAN (R 3.4.2) 
##  ggplot2      * 2.2.1   2016-12-30 cran (@2.2.1)  
##  glue           1.1.1   2017-06-21 cran (@1.1.1)  
##  graphics     * 3.4.3   2017-12-01 local          
##  grDevices    * 3.4.3   2017-12-01 local          
##  grid           3.4.3   2017-12-01 local          
##  gtable         0.2.0   2016-02-26 cran (@0.2.0)  
##  gtools         3.5.0   2015-05-29 CRAN (R 3.4.2) 
##  highr          0.6     2016-05-09 cran (@0.6)    
##  hms            0.3     2016-11-22 cran (@0.3)    
##  htmltools      0.3.6   2017-04-28 cran (@0.3.6)  
##  httr           1.3.1   2017-08-20 CRAN (R 3.4.3) 
##  kableExtra   * 0.6.1   2017-11-01 CRAN (R 3.4.2) 
##  knitr        * 1.17    2017-08-10 cran (@1.17)   
##  labeling       0.3     2014-08-23 cran (@0.3)    
##  lattice        0.20-35 2017-03-25 CRAN (R 3.3.3) 
##  lazyeval       0.2.1   2017-10-29 cran (@0.2.1)  
##  lme4         * 1.1-15  2017-12-21 cran (@1.1-15) 
##  magrittr     * 1.5     2014-11-22 cran (@1.5)    
##  MASS         * 7.3-48  2017-12-24 CRAN (R 3.4.3) 
##  Matrix       * 1.2-11  2017-08-16 CRAN (R 3.4.1) 
##  memoise        1.1.0   2017-04-21 CRAN (R 3.2.3) 
##  methods      * 3.4.3   2017-12-01 local          
##  metR         * 0.1.0   2018-01-09 local          
##  minqa          1.2.4   2014-10-09 cran (@1.2.4)  
##  munsell        0.4.3   2016-02-13 cran (@0.4.3)  
##  nlme         * 3.1-131 2017-02-06 CRAN (R 3.4.0) 
##  nloptr         1.0.4   2017-08-22 cran (@1.0.4)  
##  pkgconfig      2.0.1   2017-03-21 cran (@2.0.1)  
##  plyr           1.8.4   2016-06-08 cran (@1.8.4)  
##  purrr          0.2.4   2017-10-18 cran (@0.2.4)  
##  R6             2.2.2   2017-06-17 CRAN (R 3.2.3) 
##  RColorBrewer   1.1-2   2014-12-07 cran (@1.1-2)  
##  Rcpp           0.12.13 2017-09-28 cran (@0.12.13)
##  readr          1.1.1   2017-05-16 cran (@1.1.1)  
##  rlang          0.1.4   2017-11-05 cran (@0.1.4)  
##  rmarkdown      1.8     2017-11-17 cran (@1.8)    
##  rprojroot      1.2     2017-01-16 cran (@1.2)    
##  rvest          0.3.2   2016-06-17 CRAN (R 3.4.2) 
##  scales         0.5.0   2017-08-24 cran (@0.5.0)  
##  splines        3.4.3   2017-12-01 local          
##  stats        * 3.4.3   2017-12-01 local          
##  stringi        1.1.5   2017-04-07 cran (@1.1.5)  
##  stringr        1.2.0   2017-02-18 cran (@1.2.0)  
##  tibble         1.3.4   2017-08-22 cran (@1.3.4)  
##  tidyr        * 0.7.2   2017-10-16 cran (@0.7.2)  
##  tidyselect     0.2.2   2017-10-10 cran (@0.2.2)  
##  tools          3.4.3   2017-12-01 local          
##  utils        * 3.4.3   2017-12-01 local          
##  viridisLite    0.2.0   2017-03-24 cran (@0.2.0)  
##  withr          2.1.0   2017-11-01 cran (@2.1.0)  
##  xml2           1.1.1   2017-01-24 CRAN (R 3.4.2) 
##  yaml           2.1.15  2017-12-01 cran (@2.1.15)</code></pre>
=======
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       Europe/Warsaw               
##  date     2018-01-09</code></pre>
<pre><code>## Packages -----------------------------------------------------------------</code></pre>
<pre><code>##  package      * version date       source                        
##  assertthat     0.2.0   2017-04-11 CRAN (R 3.4.0)                
##  backports      1.1.1   2017-09-25 CRAN (R 3.4.2)                
##  base         * 3.4.3   2017-12-07 local                         
##  bindr          0.1     2016-11-13 CRAN (R 3.4.0)                
##  bindrcpp     * 0.2     2017-06-17 CRAN (R 3.4.0)                
##  colorspace     1.3-2   2016-12-14 CRAN (R 3.4.0)                
##  compiler       3.4.3   2017-12-07 local                         
##  datasets     * 3.4.3   2017-12-07 local                         
##  devtools       1.13.3  2017-08-02 CRAN (R 3.4.1)                
##  digest         0.6.13  2017-12-14 cran (@0.6.13)                
##  dplyr        * 0.7.4   2017-09-28 CRAN (R 3.4.2)                
##  evaluate       0.10.1  2017-06-24 CRAN (R 3.4.1)                
##  gdata          2.17.0  2015-07-04 CRAN (R 3.2.0)                
##  ggplot2      * 2.2.1   2016-12-30 CRAN (R 3.4.0)                
##  glue           1.1.1   2017-06-21 CRAN (R 3.4.1)                
##  graphics     * 3.4.3   2017-12-07 local                         
##  grDevices    * 3.4.3   2017-12-07 local                         
##  grid           3.4.3   2017-12-07 local                         
##  gtable         0.2.0   2016-02-26 CRAN (R 3.2.3)                
##  gtools         3.5.0   2015-05-29 CRAN (R 3.2.0)                
##  highr          0.6     2016-05-09 CRAN (R 3.4.0)                
##  hms            0.4.0   2017-11-23 cran (@0.4.0)                 
##  htmltools      0.3.6   2017-04-28 CRAN (R 3.4.0)                
##  httr           1.3.1   2017-08-20 CRAN (R 3.4.1)                
##  kableExtra   * 0.5.2   2017-09-15 cran (@0.5.2)                 
##  knitr        * 1.18    2017-12-27 cran (@1.18)                  
##  labeling       0.3     2014-08-23 CRAN (R 3.2.2)                
##  lattice        0.20-35 2017-03-25 CRAN (R 3.4.3)                
##  lazyeval       0.2.0   2016-06-12 CRAN (R 3.4.0)                
##  lme4         * 1.1-15  2017-12-21 cran (@1.1-15)                
##  magrittr     * 1.5     2014-11-22 CRAN (R 3.2.2)                
##  MASS         * 7.3-47  2017-02-26 CRAN (R 3.4.3)                
##  Matrix       * 1.2-12  2017-11-15 CRAN (R 3.4.2)                
##  memoise        1.1.0   2017-04-21 CRAN (R 3.4.0)                
##  methods      * 3.4.3   2017-12-07 local                         
##  metR         * 0.1.0   2018-01-09 local (geneticsMiNIng/metR@NA)
##  minqa          1.2.4   2014-10-09 CRAN (R 3.1.1)                
##  munsell        0.4.3   2016-02-13 CRAN (R 3.2.3)                
##  nlme         * 3.1-131 2017-02-06 CRAN (R 3.4.3)                
##  nloptr         1.0.4   2014-08-04 CRAN (R 3.2.2)                
##  pkgconfig      2.0.1   2017-03-21 CRAN (R 3.4.0)                
##  plyr           1.8.4   2016-06-08 CRAN (R 3.4.0)                
##  purrr          0.2.4   2017-10-18 cran (@0.2.4)                 
##  R6             2.2.2   2017-06-17 CRAN (R 3.4.0)                
##  RColorBrewer   1.1-2   2014-12-07 CRAN (R 3.2.2)                
##  Rcpp           0.12.14 2017-11-23 cran (@0.12.14)               
##  readr          1.1.1   2017-05-16 CRAN (R 3.4.0)                
##  rlang          0.1.6   2017-12-21 CRAN (R 3.4.3)                
##  rmarkdown      1.8     2017-11-17 cran (@1.8)                   
##  rprojroot      1.2     2017-01-16 CRAN (R 3.4.0)                
##  rvest          0.3.2   2016-06-17 CRAN (R 3.4.0)                
##  scales         0.5.0   2017-08-24 CRAN (R 3.4.1)                
##  splines        3.4.3   2017-12-07 local                         
##  stats        * 3.4.3   2017-12-07 local                         
##  stringi        1.1.6   2017-11-17 cran (@1.1.6)                 
##  stringr        1.2.0   2017-02-18 CRAN (R 3.4.0)                
##  tibble         1.3.4   2017-08-22 CRAN (R 3.4.1)                
##  tidyr        * 0.7.2   2017-10-16 cran (@0.7.2)                 
##  tidyselect     0.2.2   2017-10-10 cran (@0.2.2)                 
##  tools          3.4.3   2017-12-07 local                         
##  utils        * 3.4.3   2017-12-07 local                         
##  withr          2.1.0   2017-11-01 cran (@2.1.0)                 
##  xml2           1.1.1   2017-01-24 CRAN (R 3.4.0)                
##  yaml           2.1.16  2017-12-12 cran (@2.1.16)</code></pre>
>>>>>>> eaba4c4847b664aaeed158990fb581ecda4dedda
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#the-data">The data</a></li>
      <li><a href="#process-data">Process data</a></li>
      <li><a href="#examples-of-differentially-methylated-regions">Examples of differentially methylated regions</a></li>
      <li><a href="#comparing-all-methods">Comparing all methods</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aleksandra Brodecka, Przemyslaw Biecek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
