<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to metR • metR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">metR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/metR_vignette.html">Introduction to metR</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/geneticsMiNIng/metR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Introduction to metR</h1>
                        <h4 class="author">Aleksandra Brodecka, Przemysław Biecek</h4>
            
            <h4 class="date">2017-12-12</h4>
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><code>metR</code> package allows comprehensive data analysis from methylation studies. By <code>metR</code> we can create regions, get basic statistics about them, plot methylation rate or arrange created regions from the most interesting by different methods.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools::<span class="kw">install_github</span>(<span class="st">"geneticsMiNIng/metR"</span>)
<span class="kw">library</span>(metR)

<span class="co"># useful for printing tables</span>
<span class="kw">library</span>(kableExtra)
<span class="kw">library</span>(knitr)</code></pre></div>
<p>Initially we check if <code>metR</code> package includes all necessary datasets and functions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/check_before_using.html">check_before_using</a></span>()</code></pre></div>
<pre><code>## [1] "OK! All functions and datasets are complet in metR package"</code></pre>
<p>Ok, so let’s start!</p>
</div>
<div id="comprehensive-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#comprehensive-analysis" class="anchor"></a>Comprehensive analysis:</h1>
<p>Data for this example are downloaded from site: <a href="http://www.neuroepigenomics.org/methylomedb/download.html" class="uri">http://www.neuroepigenomics.org/methylomedb/download.html</a> We used 4 control samples: Control1 AC, Control2 AC, Control3 AC and Control4 AC and 4 disease samples: SCZ1 AC, SCZ2 AC, SCZ3 AC, SCZ4 AC. After downloading we summed all results on the same position and chromosome separately from control and disease samples.</p>
<p>In this file we created analysis of DMR in chromosome 1. The code about preprocessing files can be found in <a href="https://github.com/geneticsMiNIng/metR/blob/master/examples/prep.MethylomeDB.R" class="uri">https://github.com/geneticsMiNIng/metR/blob/master/examples/prep.MethylomeDB.R</a>. Result of this script is one data.frame <code>schizophrenia</code> that is included in metR package. Column category indicates if data are from control samples (‘control’ value) or from disease samples (‘disease’ value).</p>
<div id="examples-of-differentially-methylated-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#examples-of-differentially-methylated-regions" class="anchor"></a>Examples of differentially methylated regions</h2>
<p>We show 8 regions that are results of running metR package on schizophrenia data.</p>
<div id="region-with-maximum-coverage" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-coverage" class="anchor"></a>1. Region with maximum coverage</h3>
<p>This region has the largest coverage from interesting areas. We have 394 observations. This region hasn’t huge methylation diff in two samples but this difference is stable along the entire length.</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-6-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
923898
</td>
<td style="text-align:right;color: black;">
927566
</td>
<td style="text-align:right;color: black;">
394
</td>
<td style="text-align:right;color: black;">
0.9
</td>
<td style="text-align:right;color: black;">
0.86
</td>
<td style="text-align:right;color: black;">
0.13
</td>
<td style="text-align:right;color: black;">
0.01
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0.12
</td>
<td style="text-align:right;color: black;">
0.94
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-mean-of-methylation-rate-in-control-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-mean-of-methylation-rate-in-control-sample" class="anchor"></a>2. Region with maximum mean of methylation rate in control sample</h3>
<p>This region has the largest mean of methylation rate in control sample from interesting areas. We have 104 observations and mean of methylation rate is equal 1. This means that all of 104 observations have methylation rate on the level 1!</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-7-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
3086668
</td>
<td style="text-align:right;color: black;">
3088334
</td>
<td style="text-align:right;color: black;">
104
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.9
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.79
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.76
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.03
</td>
<td style="text-align:right;color: black;">
0.21
</td>
<td style="text-align:right;color: black;">
0.98
</td>
<td style="text-align:left;color: black;">
Ttest
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-minimum-mean-of-methylation-rate-in-disease-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-minimum-mean-of-methylation-rate-in-disease-sample" class="anchor"></a>3. Region with minimum mean of methylation rate in disease sample</h3>
<p>This region has the smallest mean of methylation rate in disease sample from interesting areas. We have 210 observations and mean of methylation rate is equal 0. This means that all of 210 observations have methylation rate on the level 0!</p>
<p>This is reverse example to the previous one.</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-8-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
20985025
</td>
<td style="text-align:right;color: black;">
20986706
</td>
<td style="text-align:right;color: black;">
210
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.05
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.01
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.06
</td>
<td style="text-align:right;color: black;">
0.85
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-standard-deviation-of-methylation-rate-in-control-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-standard-deviation-of-methylation-rate-in-control-sample" class="anchor"></a>4. Region with maximum standard deviation of methylation rate in control sample</h3>
<p>We see very interesting regions below. In this example methylation rate start from 0 in both sample and increase to 1 in control sample and 0.75 in disease sample.</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-9-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
109593468
</td>
<td style="text-align:right;color: black;">
109596927
</td>
<td style="text-align:right;color: black;">
223
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.83
</td>
<td style="text-align:right;color: black;">
0.42
</td>
<td style="text-align:right;color: black;">
0.36
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.47
</td>
<td style="text-align:right;color: black;">
0.39
</td>
<td style="text-align:right;color: black;">
0.07
</td>
<td style="text-align:right;color: black;">
0.87
</td>
<td style="text-align:left;color: black;">
Reg.Log
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-methylation-difference-in-two-probes" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-methylation-difference-in-two-probes" class="anchor"></a>5. Region with maximum methylation difference in two probes</h3>
<p>This area has only two observations but methylation difference between them is huge and equal 0.7.</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-10-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
188542753
</td>
<td style="text-align:right;color: black;">
188542830
</td>
<td style="text-align:right;color: black;">
2
</td>
<td style="text-align:right;color: black;">
0.82
</td>
<td style="text-align:right;color: black;">
0.11
</td>
<td style="text-align:right;color: black;">
0.8
</td>
<td style="text-align:right;color: black;">
0.11
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.78
</td>
<td style="text-align:right;color: black;">
0.1
</td>
<td style="text-align:right;color: black;">
0.03
</td>
<td style="text-align:right;color: black;">
0.01
</td>
<td style="text-align:right;color: black;">
0.7
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:left;color: black;">
Reg.Mixed.Beta
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-rank-function" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-rank-function" class="anchor"></a>6. Region with maximum rank function</h3>
<p>This area has maximum rank function. Rank function is based on quantile regression which uses information about number of observations in region. In this example methylation difference between probes is the biggest with respect to number of observations. Methylation difference is close to methylation difference in previous example but has more observation (11 to 2).</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-11-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
79270449
</td>
<td style="text-align:right;color: black;">
79270840
</td>
<td style="text-align:right;color: black;">
11
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.31
</td>
<td style="text-align:right;color: black;">
0.98
</td>
<td style="text-align:right;color: black;">
0.31
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.93
</td>
<td style="text-align:right;color: black;">
0.31
</td>
<td style="text-align:right;color: black;">
0.03
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.66
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:left;color: black;">
Reg.Log.Beta
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-methylation-range-in-control-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-methylation-range-in-control-sample" class="anchor"></a>7. Region with maximum methylation range in control sample</h3>
<p>This region has the biggest methylation range in control sample from interesting areas. We see that methylation rate in control sample starts from 1 and decreases to 0. In disease sample methylation rate starts from level 0.9 and decreases to 0.15. Curious observations is occurence that at the beginning of region we see bigger methylation in control sample and at the end of region reverse case.</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-12-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
62673335
</td>
<td style="text-align:right;color: black;">
62676253
</td>
<td style="text-align:right;color: black;">
233
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.93
</td>
<td style="text-align:right;color: black;">
0.13
</td>
<td style="text-align:right;color: black;">
0.25
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.14
</td>
<td style="text-align:right;color: black;">
0.29
</td>
<td style="text-align:right;color: black;">
0.22
</td>
<td style="text-align:right;color: black;">
0.12
</td>
<td style="text-align:right;color: black;">
0.93
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
<div id="region-with-maximum-methylation-range-in-disease-sample" class="section level3">
<h3 class="hasAnchor">
<a href="#region-with-maximum-methylation-range-in-disease-sample" class="anchor"></a>8. Region with maximum methylation range in disease sample</h3>
<p>This region has the biggest methylation range in disease sample from interesting areas. In the middle of region we can notice that methylation rate in disease sample is equal to 0 and in control is around 0.25. At the beginning and at the end of regions methylation increases in two probes.</p>
<img src="metR_vignette_files/figure-html/unnamed-chunk-13-1.png" width="720"><table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
39342909
</td>
<td style="text-align:right;color: black;">
39345135
</td>
<td style="text-align:right;color: black;">
177
</td>
<td style="text-align:right;color: black;">
0.94
</td>
<td style="text-align:right;color: black;">
0.95
</td>
<td style="text-align:right;color: black;">
0.3
</td>
<td style="text-align:right;color: black;">
0.06
</td>
</tr></tbody>
</table>
<table class="table" style="font-size: 11px; ">
<thead><tr>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
<th style="text-align:left;">
source
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:right;color: black;">
0.24
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.12
</td>
<td style="text-align:right;color: black;">
0.19
</td>
<td style="text-align:right;color: black;">
0.24
</td>
<td style="text-align:right;color: black;">
0.99
</td>
<td style="text-align:left;color: black;">
Wilcoxon
</td>
</tr></tbody>
</table>
</div>
</div>
<div id="comparing-all-methods" class="section level2">
<h2 class="hasAnchor">
<a href="#comparing-all-methods" class="anchor"></a>Comparing all methods</h2>
<p>We use 8 methods which are available in metR package:</p>
<ul>
<li><p>Ttest - t-Student test on methylation.rate</p></li>
<li><p>Wilcoxon - Wilcoxon test on methylation.rate</p></li>
<li><p>KS - Kolmogorov-Smirnov test on methylation.rate</p></li>
</ul>
<p>Regression methods, where number of success are number of methylated citosines :</p>
<ul>
<li><p>Reg.Log - standard logistic regression</p></li>
<li><p>Reg.Mixed - logistic regression where random effects are positions of chromosome</p></li>
<li><p>Reg.Corr.Mixed - logistic regression where random effects are positions of chromosome and correlation matrix between position is previously estimated and included in <code>metR</code> package</p></li>
</ul>
<p>These methods sorts regions based on p.value of grouping variable. Methods Reg.Log.Beta, Reg.Mixed.Beta, Reg.Corr.Mixed.Beta order regions based on beta coefficients of grouping variable.</p>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-14-1.png" width="720"></p>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-15-1.png" width="720"></p>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-16-1.png" width="720"></p>
<p>We can show, that these method given very different results. The rank rate is very good if we use Ttest, Reg.Log, Reg.Mixed.Beta and Reg.Log.Beta methods. The biggest differences we see for methylation coverage. Wilcoxon test recommended regions that have the most observations ~ 200. Reg.Corr.Mixed.Beta and Reg.Log.Beta proposed smaller group - about 10 observations. If we check methylation difference, Reg.Log.Beta and Reg.Mixed.Beta present very large difference in recommended regions.</p>
</div>
</div>
<div id="how-we-obtain-these-results" class="section level1">
<h1 class="hasAnchor">
<a href="#how-we-obtain-these-results" class="anchor"></a>How we obtain these results?</h1>
<div id="preprocess-data" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocess-data" class="anchor"></a>1. Preprocess data:</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(metR)

<span class="kw">data</span>(<span class="st">'schizophrenia'</span>)
control &lt;-<span class="st"> </span>schizophrenia %&gt;%<span class="st"> </span><span class="kw">filter</span>(category ==<span class="st"> 'control'</span>) %&gt;%
<span class="st">  </span>dplyr::<span class="kw">select</span>(-category)

disease &lt;-<span class="st"> </span>schizophrenia %&gt;%<span class="st"> </span><span class="kw">filter</span>(category ==<span class="st"> 'disease'</span>) %&gt;%
<span class="st">  </span>dplyr::<span class="kw">select</span>(-category)</code></pre></div>
<p><code>preprocessing</code> function need two probes with specific columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(control, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) %&gt;%<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span>:<span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73465
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73467
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
74406
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
</tr>
</tbody>
</table>
<p>So:</p>
<ul>
<li><p>chr - chromosome name</p></li>
<li><p>poz - position in chromosome</p></li>
<li><p>no - number of reads per position</p></li>
<li><p>meth - number of reads with methylated cytosine</p></li>
<li><p>unmeth - number of reads with unmethylated cytosine</p></li>
<li><p>meth.rate - ratio: meth/(meth + unmeth)</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/preprocessing.html">preprocessing</a></span>(control, disease)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(data, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) %&gt;%<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span>:<span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>) %&gt;%
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/column_spec">column_spec</a></span>(<span class="dv">3</span>, <span class="dt">background =</span> <span class="st">"#f21a1a"</span>, <span class="dt">color =</span> <span class="st">'white'</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:left;">
prob
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;color: white;background-color: #f21a1a;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;color: white;background-color: #f21a1a;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;color: white;background-color: #f21a1a;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
</tr>
</tbody>
</table>
<p>By <code>preprocessing</code> we get one data frame where we have both results of methylation in position and chromosome from control and disease samples. Prob columns indicates if data are from control samples (x), or from disease samples (y).</p>
</div>
<div id="creating-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-regions" class="anchor"></a>2. Creating regions</h2>
<p>After preprocessing we can create region. After creation they will be tested if there is significant difference between two probes in methylation rate.</p>
<p>We use function <code>create_tiles_max_gap</code> with argument <code>gaps.length = 100</code>. The same regions are observations that maximum difference position is gaps.length argument.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles<span class="fl">.3</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_tiles_max_gap.html">create_tiles_max_gap</a></span>(data, <span class="dt">gaps.length =</span> <span class="dv">100</span>)</code></pre></div>
<p>Column tiles indicates id of region in chromosome:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(data.tiles<span class="fl">.3</span>, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) %&gt;%<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span>:<span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>)  %&gt;%
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/column_spec">column_spec</a></span>(<span class="dv">9</span>, <span class="dt">background =</span> <span class="st">"#f21a1a"</span>, <span class="dt">color =</span> <span class="st">'white'</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:left;">
prob
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
<th style="text-align:right;">
tiles
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1.1
</td>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
1
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1.2
</td>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
1
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1.3
</td>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
1
</td>
</tr>
</tbody>
</table>
<p>It’s also possible to use <code>create_tiles_fixed_length function</code>, where <code>tiles.length</code> that specifies maximum difference between minimum and maximum position in the same methylation regions. If <code>common = TRUE</code> function creates second regions group that are min position is (min position + max position)/2 of k-region and max position is (min position + max position) of k+1 region.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.tiles<span class="fl">.1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_tiles_fixed_length.html">create_tiles_fixed_length</a></span>(data, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> F)
data.tiles<span class="fl">.2</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_tiles_fixed_length.html">create_tiles_fixed_length</a></span>(data, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> T)</code></pre></div>
</div>
<div id="get-basic-statistics" class="section level2">
<h2 class="hasAnchor">
<a href="#get-basic-statistics" class="anchor"></a>3. Get basic statistics</h2>
<p>We get basic statistics about two probes by <code>get_stats</code> function. This is helpful if we want check coverage of created regions or methylation difference.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">stats<span class="fl">.3</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_stats.html">get_stats</a></span>(data.tiles<span class="fl">.3</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(stats<span class="fl">.3</span>, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>) %&gt;%<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span>:<span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>) </code></pre></div>
<table class="table" style="font-size: 9px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:right;color: black;">
2
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.0398496
</td>
<td style="text-align:right;color: black;">
0.4672649
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81546
</td>
<td style="text-align:right;color: black;">
81546
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
NA
</td>
<td style="text-align:right;color: black;">
NA
</td>
<td style="text-align:right;color: black;">
0.0398496
</td>
<td style="text-align:right;color: black;">
0.4051596
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81698
</td>
<td style="text-align:right;color: black;">
81863
</td>
<td style="text-align:right;color: black;">
4
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.0398496
</td>
<td style="text-align:right;color: black;">
0.5333629
</td>
</tr>
</tbody>
</table>
<p>We get basic statistics about each region:</p>
<ul>
<li><p>minimum and maximum position (start and end columns)</p></li>
<li><p>min, max, std of methylation.rate in each probe (_x from control sample and _y from disease sample)</p></li>
<li><p>meth.diff - absolute difference between mean of methylation rate in control and disease sample</p></li>
<li><p>quantile - rank function based on quantile regression which uses information about number of observations in region.</p></li>
</ul>
<p>We also can join e.g stats.3 and data.tiles.3 on chromosome, start and end column and analyzing regions only if they coverage is greater than some specific values or other condition.</p>
</div>
<div id="finding-dmr" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-dmr" class="anchor"></a>4. Finding DMR</h2>
<p>We get interesting regions by using <code>find_DMR</code> function. <code>Data</code> argument is data.frame from <code>create_tiles function</code>. In methods argument we can type methods which we want to use.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># eval = F</span>
result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find_DMR.html">find_DMR</a></span>(data.tiles<span class="fl">.3</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Wilcoxon'</span>, <span class="st">'Ttest'</span>, <span class="st">'KS'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Mixed'</span>, <span class="st">'Reg.Corr.Mixed'</span>))</code></pre></div>
<p>Results of running function above are included in package. So now we only load data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">'find.DMR.results'</span>)
<span class="kw">names</span>(find.DMR.results)</code></pre></div>
<pre><code>## [1] "Wilcoxon"       "Ttest"          "KS"             "Reg.Log"       
## [5] "Reg.Mixed"      "Reg.Corr.Mixed"</code></pre>
<p>This is a list of data.frames. One data.frame is result of running specific methods.</p>
<p>If we want also get results from sorting by beta coefficient we can run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find_DMR.html">find_DMR</a></span>(data.tiles<span class="fl">.3</span>, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Reg.Log'</span>, <span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'Reg.Corr.Mixed'</span>)
                    , <span class="dt">p.value.log.reg =</span> <span class="fl">0.001</span>, <span class="dt">p.value.reg.mixed =</span> <span class="fl">0.001</span>, <span class="dt">p.value.reg.corr.mixed =</span> <span class="fl">0.001</span>)</code></pre></div>
<p>It’s better when we use results previously obtained from sorting by p.value of grouping variable because we can spare computing time:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results$Reg.Log.Beta &lt;-<span class="st"> </span>find.DMR.results$Reg.Log %&gt;%<span class="st"> </span><span class="kw">filter</span>(p.value &lt;<span class="st"> </span><span class="fl">0.001</span>) %&gt;%<span class="st"> </span><span class="kw">arrange</span>(-<span class="kw">abs</span>(beta.coef))

find.DMR.results$Reg.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results$Reg.Mixed %&gt;%<span class="st"> </span><span class="kw">filter</span>(p.value &lt;<span class="st"> </span><span class="fl">0.001</span>) %&gt;%<span class="st"> </span><span class="kw">arrange</span>(-<span class="kw">abs</span>(beta.coef))

find.DMR.results$Reg.Corr.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results$Reg.Corr.Mixed %&gt;%<span class="st"> </span><span class="kw">filter</span>(p.value &lt;<span class="st"> </span><span class="fl">0.001</span>) %&gt;%<span class="st"> </span><span class="kw">arrange</span>(-<span class="kw">abs</span>(beta.coef))</code></pre></div>
<p>Now we can select top 100 regions by each method:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get_top &lt;-<span class="st"> </span>function(data, n, stats){
  data %&gt;%<span class="st"> </span><span class="kw">slice</span>(<span class="dv">1</span>:n) %&gt;%
<span class="st">    </span><span class="kw">left_join</span>(stats, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">'chr'</span>, <span class="st">'start'</span>, <span class="st">'end'</span>)) %&gt;%
<span class="st">    </span>dplyr::<span class="kw">select</span>(chr, start, end, meth.cov, meth.max_x, meth.max_y  
                  ,meth.mean_x, meth.mean_y, meth.min_x,
                  meth.min_y, meth.sd_x, meth.sd_y, meth.diff,quantile)
}


top &lt;-<span class="st"> </span><span class="kw">do.call</span>(gdata::combine,<span class="kw">lapply</span>(find.DMR.results, get_top, <span class="dt">n =</span> <span class="dv">100</span>, <span class="dt">stats =</span> stats<span class="fl">.3</span>))</code></pre></div>
<p>In subsection ‘Comparing all methods’ data from <code>top</code> were plotted.</p>
</div>
<div id="plotting-methylation" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-methylation" class="anchor"></a>4. Plotting methylation</h2>
<p>Region in this subsection was plotted in Examples of differentially methylated regions above.</p>
<p>We use <code>draw_methylation</code> function with basic arguments:</p>
<ul>
<li><p>data - data.frame from <code>preprocessing</code> function or data.frame from <code>create_tiles_max_gap</code> or <code>create_tiles_fixed_length</code> functions</p></li>
<li><p>chr - chromosome name of region that will be plotted</p></li>
<li><p>start - min position of region that will be plotted</p></li>
<li><p>end - max position of region that will be plotted</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">i =<span class="st"> </span><span class="kw">which.max</span>(top$quantile)
<span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(data, top$chr[i], top$start[i], top$end[i], <span class="dt">bind.probes =</span> F,
        <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'none'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-30-1.png" width="720"></p>
<p>Change <code>legend.position</code> to get legend of logarithm reads in each position:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(data, top$chr[i], top$start[i], top$end[i], <span class="dt">bind.probes =</span> F,
        <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'bottom'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>, <span class="dt">legend.text =</span> <span class="dv">7</span>, <span class="dt">legend.title =</span> <span class="dv">8</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-31-1.png" width="720"></p>
<p>Change <code>bind.probes</code> to TRUE argument if you want binding each observations between two probes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(data, top$chr[i], top$start[i], top$end[i], <span class="dt">bind.probes =</span> T,
                 <span class="dt">smooth.methylation =</span> T,
  <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'none'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-32-1.png" width="720"></p>
<p>Change <code>smooth.methylation</code> to FALSE if you don’t want smoothing observations:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(data, top$chr[i], top$start[i], top$end[i], <span class="dt">bind.probes =</span> T,<span class="dt">smooth.methylation =</span> F,
        <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'none'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>)</code></pre></div>
<p><img src="metR_vignette_files/figure-html/unnamed-chunk-33-1.png" width="720"></p>
<p>Other arguments are responsible for size of element on plots.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#comprehensive-analysis">Comprehensive analysis:</a><ul class="nav nav-pills nav-stacked">
<li><a href="#examples-of-differentially-methylated-regions">Examples of differentially methylated regions</a></li>
      <li><a href="#comparing-all-methods">Comparing all methods</a></li>
      </ul>
</li>
      <li>
<a href="#how-we-obtain-these-results">How we obtain these results?</a><ul class="nav nav-pills nav-stacked">
<li><a href="#preprocess-data">1. Preprocess data:</a></li>
      <li><a href="#creating-regions">2. Creating regions</a></li>
      <li><a href="#get-basic-statistics">3. Get basic statistics</a></li>
      <li><a href="#finding-dmr">4. Finding DMR</a></li>
      <li><a href="#plotting-methylation">4. Plotting methylation</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aleksandra Brodecka, Przemyslaw Biecek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
