<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>metR: step by step • metR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top navbar-mi2" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
	  <a class="navbar-mi2logo" href="http://mi2.mini.pw.edu.pl/">
        <img src="https://github.com/mi2-warsaw/MI2template/blob/master/inst/pkgdown/assets/MI2logo.jpg?raw=true" alt="MI2" height="46" title="MI2"></a>
      <a class="navbar-brand navbar-mi2" href="../index.html">metR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="navbar-mi2 nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/metR_comparison.html">metR: a deeper study</a>
    </li>
    <li>
      <a href="../articles/metR_step_by_step.html">metR: step by step</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/geneticsMiNIng/metR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>metR: step by step</h1>
                        <h4 class="author">Aleksandra Brodecka</h4>
            
            <h4 class="date">2018-01-09</h4>
          </div>

    
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <code>metR</code> package contains tools for comprehensive analysis of data from methylation studies. With the <code>metR</code> one can define regions, get basic statistics about methylation in them, plot methylation rate or rank these regions from the most interesting to the least one.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"geneticsMiNIng/metR"</span>)
<span class="kw">library</span>(metR)
<span class="kw">library</span>(kableExtra)
<span class="kw">library</span>(knitr)</code></pre></div>
</div>
<div id="the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#the-data" class="anchor"></a>The data</h1>
<p>Data for this example are downloaded from the site: <a href="http://www.neuroepigenomics.org/methylomedb/download.html" class="uri">http://www.neuroepigenomics.org/methylomedb/download.html</a>. We used 4 control samples: Control1 AC, Control2 AC, Control3 AC and Control4 AC and 4 disease samples: SCZ1 AC, SCZ2 AC, SCZ3 AC, SCZ4 AC.</p>
<p>In the <code>metR</code> package these samples are available in the <code>schizophrenia</code> data frame. Each row is a single methylation probe. Data from all control/disease samples are combined together.</p>
<p>To speed up calculations and shrink down the size of data we are using only probes from chromosome 1. The R code needed to download data from all samples can be found in the script <a href="https://github.com/geneticsMiNIng/metR/blob/master/examples/prep.MethylomeDB.R" class="uri">https://github.com/geneticsMiNIng/metR/blob/master/examples/prep.MethylomeDB.R</a>. The result of this script is a single data.frame <code>schizophrenia</code> that is included in metR package. The last column <code>category</code> indicates if data are from control samples (‘control’) or from disease samples (‘disease’).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">'schizophrenia'</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(schizophrenia, <span class="dv">6</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>, <span class="dt">color =</span> <span class="st">"black"</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
<th style="text-align:left;">
category
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73465
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73467
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
74406
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
74873
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
75268
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
75390
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
<td style="text-align:left;color: black;">
control
</td>
</tr>
</tbody>
</table>
<div id="preprocessing" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>1. Preprocessing</h2>
<p>The data requires some preprocessing. This can be done with the <code><a href="../reference/preprocessing.html">preprocessing()</a></code> function. In this function we only select data from two given data frames if methylation probe exists in both samples on the same chromosome and position.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">control &lt;-<span class="st"> </span>schizophrenia <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(category <span class="op">==</span><span class="st"> 'control'</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>category)

disease &lt;-<span class="st"> </span>schizophrenia <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(category <span class="op">==</span><span class="st"> 'disease'</span>) <span class="op">%&gt;%</span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw"><a href="http://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>category)

processed_schizophrenia &lt;-<span class="st"> </span><span class="kw"><a href="../reference/preprocessing.html">preprocessing</a></span>(control, disease)
<span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(processed_schizophrenia, <span class="dv">6</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>, <span class="dt">color =</span> <span class="st">"black"</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:left;">
prob
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81546
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81546
</td>
<td style="text-align:left;color: black;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
</tr>
</tbody>
</table>
<p>The <code><a href="../reference/preprocessing.html">preprocessing()</a></code> function need two probes with specific columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(control, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73465
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
73467
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.1666667
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
74406
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
5
</td>
<td style="text-align:right;color: black;">
0.5000000
</td>
</tr>
</tbody>
</table>
<p>So:</p>
<ul>
<li><p>chr - chromosome name</p></li>
<li><p>poz - position in chromosome</p></li>
<li><p>no - number of reads per position</p></li>
<li><p>meth - number of reads with methylated cytosine</p></li>
<li><p>unmeth - number of reads with unmethylated cytosine</p></li>
<li><p>meth.rate - ratio: meth/(meth + unmeth)</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(processed_schizophrenia, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/column_spec">column_spec</a></span>(<span class="dv">3</span>, <span class="dt">background =</span> <span class="st">"#f21a1a"</span>, <span class="dt">color =</span> <span class="st">'white'</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:left;">
prob
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;color: white;background-color: #f21a1a;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;color: white;background-color: #f21a1a;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;color: white;background-color: #f21a1a;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
</tr>
</tbody>
</table>
<p>By <code>preprocessing</code> we get one data frame where we have both results of methylation in position and chromosome from control and disease samples. Prob columns indicates if data are from control samples (x), or from disease samples (y).</p>
</div>
</div>
<div id="tiles-regions" class="section level1">
<h1 class="hasAnchor">
<a href="#tiles-regions" class="anchor"></a>Tiles / regions</h1>
<p>All statistics in the metR package are calculated for regions of DNA. Currently the <code>metR</code> package implements two functions that defines regions:</p>
<ul>
<li>
<code><a href="../reference/create_tiles_max_gap.html">create_tiles_max_gap()</a></code> creates list of regions rich in methylation probes. Regions are defined as probes spaced not farther than <code>gaps.length</code>,</li>
<li>
<code><a href="../reference/create_tiles_fixed_length.html">create_tiles_fixed_length()</a></code> creates list of regions of fixed length <code>tiles.length</code>.</li>
</ul>
<p>After regions creation they will be tested if there is significant difference between two probes in methylation rate.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tiles_schizophrenia &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_tiles_max_gap.html">create_tiles_max_gap</a></span>(processed_schizophrenia, <span class="dt">gaps.length =</span> <span class="dv">100</span>)</code></pre></div>
<p>Column tiles indicates id of region in chromosome:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(tiles_schizophrenia, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>)  <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/column_spec">column_spec</a></span>(<span class="dv">9</span>, <span class="dt">background =</span> <span class="st">"#f21a1a"</span>, <span class="dt">color =</span> <span class="st">'white'</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:left;">
prob
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
<th style="text-align:right;">
tiles
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1.1
</td>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
1
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1.2
</td>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
1
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1.3
</td>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
1
</td>
</tr>
</tbody>
</table>
<p>It’s also possible to use <code>create_tiles_fixed_length function</code>, where <code>tiles.length</code> that specifies maximum difference between minimum and maximum position in the same methylation regions. If <code>common = TRUE</code> function creates second regions group that are min position is (min position + max position)/2 of k-region and max position is (min position + max position) of k+1 region.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tiles_schizophrenia_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_tiles_fixed_length.html">create_tiles_fixed_length</a></span>(processed_schizophrenia, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> F)
tiles_schizophrenia_<span class="dv">3</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create_tiles_fixed_length.html">create_tiles_fixed_length</a></span>(processed_schizophrenia, <span class="dt">tiles.length =</span> <span class="dv">1000</span>, <span class="dt">common =</span> T)</code></pre></div>
<p>In output from <code>create_tiles_fixed_length function</code> column tiles also indicates id of region in chromosome and tiles.common indicates second group id region:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(tiles_schizophrenia_<span class="dv">2</span>, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/column_spec">column_spec</a></span>(<span class="dv">8</span>, <span class="dt">background =</span> <span class="st">"#f21a1a"</span>, <span class="dt">color =</span> <span class="st">'white'</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:left;">
prob
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
<th style="text-align:right;">
tiles
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(tiles_schizophrenia_<span class="dv">3</span>, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>, <span class="dt">full_width =</span> F) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>)  <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/column_spec">column_spec</a></span>(<span class="dv">8</span><span class="op">:</span><span class="dv">9</span>, <span class="dt">background =</span> <span class="st">"#f21a1a"</span>, <span class="dt">color =</span> <span class="st">'white'</span>)</code></pre></div>
<table class="table" style="font-size: 9px; width: auto !important; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
poz
</th>
<th style="text-align:left;">
prob
</th>
<th style="text-align:right;">
no
</th>
<th style="text-align:right;">
meth
</th>
<th style="text-align:right;">
unmeth
</th>
<th style="text-align:right;">
meth.rate
</th>
<th style="text-align:right;">
tiles
</th>
<th style="text-align:right;">
tiles.common
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:left;color: black;">
y
</td>
<td style="text-align:right;color: black;">
76
</td>
<td style="text-align:right;color: black;">
66
</td>
<td style="text-align:right;color: black;">
10
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:left;color: black;">
x
</td>
<td style="text-align:right;color: black;">
35
</td>
<td style="text-align:right;color: black;">
29
</td>
<td style="text-align:right;color: black;">
6
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
<td style="text-align:right;color: black;color: white;background-color: #f21a1a;">
81
</td>
</tr>
</tbody>
</table>
</div>
<div id="basic-statistics" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-statistics" class="anchor"></a>Basic statistics</h1>
<p>We get basic statistics about two probes by <code>get_stats</code> function. This is helpful if we want check coverage of created regions or methylation difference.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">stats &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_stats.html">get_stats</a></span>(tiles_schizophrenia)
<span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/kable_styling">kable_styling</a></span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(stats, <span class="dv">3</span>), <span class="st">"html"</span>), <span class="dt">position =</span> <span class="st">"left"</span>,
<span class="dt">font_size =</span> <span class="dv">9</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/kableExtra/topics/row_spec">row_spec</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dt">color =</span> <span class="st">"black"</span>) </code></pre></div>
<table class="table" style="font-size: 9px; ">
<thead><tr>
<th style="text-align:left;">
chr
</th>
<th style="text-align:right;">
start
</th>
<th style="text-align:right;">
end
</th>
<th style="text-align:right;">
meth.cov
</th>
<th style="text-align:right;">
meth.max_x
</th>
<th style="text-align:right;">
meth.max_y
</th>
<th style="text-align:right;">
meth.mean_x
</th>
<th style="text-align:right;">
meth.mean_y
</th>
<th style="text-align:right;">
meth.min_x
</th>
<th style="text-align:right;">
meth.min_y
</th>
<th style="text-align:right;">
meth.sd_x
</th>
<th style="text-align:right;">
meth.sd_y
</th>
<th style="text-align:right;">
meth.diff
</th>
<th style="text-align:right;">
quantile
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81412
</td>
<td style="text-align:right;color: black;">
81442
</td>
<td style="text-align:right;color: black;">
2
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.0398496
</td>
<td style="text-align:right;color: black;">
0.4672649
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81546
</td>
<td style="text-align:right;color: black;">
81546
</td>
<td style="text-align:right;color: black;">
1
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
NA
</td>
<td style="text-align:right;color: black;">
NA
</td>
<td style="text-align:right;color: black;">
0.0398496
</td>
<td style="text-align:right;color: black;">
0.4051596
</td>
</tr>
<tr>
<td style="text-align:left;color: black;">
chr1
</td>
<td style="text-align:right;color: black;">
81698
</td>
<td style="text-align:right;color: black;">
81863
</td>
<td style="text-align:right;color: black;">
4
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0.8285714
</td>
<td style="text-align:right;color: black;">
0.8684211
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0
</td>
<td style="text-align:right;color: black;">
0.0398496
</td>
<td style="text-align:right;color: black;">
0.5333629
</td>
</tr>
</tbody>
</table>
<p>We get basic statistics about each region:</p>
<ul>
<li><p>minimum and maximum position (start and end columns)</p></li>
<li><p>min, max, std of methylation.rate in each probe (_x from control sample and _y from disease sample)</p></li>
<li><p>meth.diff - absolute difference between mean of methylation rate in control and disease sample</p></li>
<li><p>quantile - rank function based on quantile regression which uses information about number of observations in region.</p></li>
</ul>
<p>We also can join e.g stats and tiles_schizophrenia on chromosome, start and end column and analyzing regions only if they coverage is greater than some specific values or other condition.</p>
</div>
<div id="finding-dmr" class="section level1">
<h1 class="hasAnchor">
<a href="#finding-dmr" class="anchor"></a>Finding DMR</h1>
<p>We get interesting regions by using <code>find_DMR</code> function. <code>Data</code> argument is data.frame from <code>create_tiles_fixed_length function</code> or from <code>create_tiles_max_gap function</code> . In methods argument we can type methods which we want to use:</p>
<p>We use 8 methods which are available in metR package:</p>
<p>In following method are compared methylation rate between x and y prob on the same position and chromosome. These methods sorts regions based on p.value from adequate test results:</p>
<ul>
<li><p><code>Ttest</code> - t-test with unequal variance with null hypothesis that mean of methylation rate is equal in two probes.</p></li>
<li><p><code>Wilcoxon</code> - Wilcoxon signed test with null hypothesis that median of methylation rate is equal in two probes.</p></li>
<li><p><code>KS</code> - Kolmogorov-Smirnov test with null hypothesis that distribution of methylation rate is equal in two probes. We use two sided alternative hypothesis.</p></li>
</ul>
<p>In <code>Ttest</code>, <code>Wilcoxon</code> and <code>KS</code> test we use two sided alternative hypothesis.</p>
<p>We also implemented regression methods, where number of success are number of methylated citosines and failures are number of unmethylated citosines. Output from this methods is beta coefficient of indicator variable from regression model and criticial value from Wald test on indicator variable. Indicator variable is equal 1 if observations are from x prob and 0 otherwise.</p>
<ul>
<li><p><code>Reg.Log</code> - standard logistic regression where explantatory variables are positions of chromosome and indicator if observations are from x prob.</p></li>
<li><p><code>Reg.Mixed</code> - logistic regression with random effects. In this method explantatory variable is only indicator if observations are from x prob and positions of chromosome are random effects.</p></li>
<li><p><code>Reg.Corr.Mixed</code> - logistic regression with random effects with given correlation matrix. In this method explantatory variable is only indicator if observations are from x prob and positions of chromosome are random effects. Correlation matrix between position is previously estimated and included in <code>metR</code> package.</p></li>
</ul>
<p>Methods <code>Reg.Log.Beta</code>, <code>Reg.Mixed.Beta</code>, <code>Reg.Corr.Mixed.Beta</code> order regions based on beta coefficients of grouping variable or p.values of grouping variable.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># eval = F</span>
result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find_DMR.html">find_DMR</a></span>(tiles_schizophrenia, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Wilcoxon'</span>, <span class="st">'Ttest'</span>, <span class="st">'KS'</span>, <span class="st">'Reg.Log'</span>, <span class="st">'Reg.Mixed'</span>, <span class="st">'Reg.Corr.Mixed'</span>))</code></pre></div>
<p>Results of running function above are included in package. So now we only load data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">'find.DMR.results'</span>)
<span class="kw">names</span>(find.DMR.results)</code></pre></div>
<pre><code>## [1] "Wilcoxon"       "Ttest"          "KS"             "Reg.Log"       
## [5] "Reg.Mixed"      "Reg.Corr.Mixed"</code></pre>
<p>This is a list of data.frames. One data.frame is result of running specific methods. If we want also get results from sorting by beta coefficient we can run:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/find_DMR.html">find_DMR</a></span>(tiles_schizophrenia, <span class="dt">methods =</span> <span class="kw">c</span>(<span class="st">'Reg.Log'</span>, <span class="st">'Reg.Corr.Mixed'</span>, <span class="st">'Reg.Corr.Mixed'</span>)
                    , <span class="dt">p.value.log.reg =</span> <span class="fl">0.001</span>, <span class="dt">p.value.reg.mixed =</span> <span class="fl">0.001</span>, <span class="dt">p.value.reg.corr.mixed =</span> <span class="fl">0.001</span>)</code></pre></div>
<p>It’s better when we use results previously obtained from sorting by p.value of grouping variable because we can spare computing time:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">find.DMR.results<span class="op">$</span>Reg.Log.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Log <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))

find.DMR.results<span class="op">$</span>Reg.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Mixed <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))

find.DMR.results<span class="op">$</span>Reg.Corr.Mixed.Beta &lt;-<span class="st"> </span>find.DMR.results<span class="op">$</span>Reg.Corr.Mixed <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(p.value <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.001</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="op">-</span><span class="kw">abs</span>(beta.coef))</code></pre></div>
<p>Now we can select top 100 regions by each method using <code>get_top</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">top &lt;-<span class="st"> </span><span class="kw">do.call</span>(gdata<span class="op">::</span>combine,<span class="kw">lapply</span>(find.DMR.results, get_top, <span class="dt">n =</span> <span class="dv">100</span>, <span class="dt">stats =</span> stats))</code></pre></div>
</div>
<div id="plotting-methylation" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-methylation" class="anchor"></a>Plotting methylation</h1>
<p>We use <code>draw_methylation</code> function with basic arguments:</p>
<ul>
<li><p>data - data.frame from <code>preprocessing</code> function or data.frame from <code>create_tiles_max_gap</code> or <code>create_tiles_fixed_length</code> functions</p></li>
<li><p>chr - chromosome name of region that will be plotted</p></li>
<li><p>start - min position of region that will be plotted</p></li>
<li><p>end - max position of region that will be plotted</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">i =<span class="st"> </span><span class="kw">which.max</span>(top<span class="op">$</span>quantile)
<span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(processed_schizophrenia, top<span class="op">$</span>chr[i], top<span class="op">$</span>start[i], top<span class="op">$</span>end[i], <span class="dt">bind.probes =</span> F,
        <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'none'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>)</code></pre></div>
<p><img src="metR_step_by_step_files/figure-html/unnamed-chunk-17-1.png" width="720"></p>
<p>Change <code>legend.position</code> to get legend of logarithm reads in each position:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(processed_schizophrenia, top<span class="op">$</span>chr[i], top<span class="op">$</span>start[i], top<span class="op">$</span>end[i], <span class="dt">bind.probes =</span> F,
        <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'bottom'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>, <span class="dt">legend.text =</span> <span class="dv">7</span>, <span class="dt">legend.title =</span> <span class="dv">8</span>)</code></pre></div>
<p><img src="metR_step_by_step_files/figure-html/unnamed-chunk-18-1.png" width="720"></p>
<p>Change <code>bind.probes</code> to TRUE argument if you want binding each observations between two probes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(processed_schizophrenia, top<span class="op">$</span>chr[i], top<span class="op">$</span>start[i], top<span class="op">$</span>end[i], <span class="dt">bind.probes =</span> T,
                 <span class="dt">smooth.methylation =</span> T,
  <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'none'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>)</code></pre></div>
<p><img src="metR_step_by_step_files/figure-html/unnamed-chunk-19-1.png" width="720"></p>
<p>Change <code>smooth.methylation</code> to FALSE if you don’t want smoothing observations:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/draw_methylation.html">draw_methylation</a></span>(processed_schizophrenia, top<span class="op">$</span>chr[i], top<span class="op">$</span>start[i], top<span class="op">$</span>end[i], <span class="dt">bind.probes =</span> T,<span class="dt">smooth.methylation =</span> F,
        <span class="dt">size.x.dot =</span> <span class="fl">3.5</span>, <span class="dt">size.y.dot =</span> <span class="dv">2</span>, <span class="dt">plot.title =</span> <span class="dv">13</span>, <span class="dt">axis.title.x =</span> <span class="dv">12</span>, <span class="dt">axis.title.y =</span> <span class="dv">12</span>, <span class="dt">legend.position =</span> <span class="st">'none'</span>,
        <span class="dt">axis.text.x =</span> <span class="dv">9</span>, <span class="dt">axis.text.y =</span> <span class="dv">9</span>)</code></pre></div>
<p><img src="metR_step_by_step_files/figure-html/unnamed-chunk-20-1.png" width="720"></p>
<p>Other arguments are responsible for size of element on plots.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/session_info">session_info</a></span>()</code></pre></div>
<pre><code>## Session info -------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.4.3 (2017-11-30)
##  system   x86_64, darwin15.6.0        
##  ui       X11                         
<<<<<<< HEAD
##  language pl_PL                       
##  collate  pl_PL.UTF-8                 
##  tz       Poland                      
##  date     2018-01-09</code></pre>
<pre><code>## Packages -----------------------------------------------------------------</code></pre>
<pre><code>##  package     * version date       source         
##  assertthat    0.2.0   2017-04-11 cran (@0.2.0)  
##  backports     1.1.1   2017-09-25 cran (@1.1.1)  
##  base        * 3.4.3   2017-12-01 local          
##  bindr         0.1     2016-11-13 cran (@0.1)    
##  bindrcpp    * 0.2     2017-06-17 cran (@0.2)    
##  colorspace    1.3-2   2016-12-14 cran (@1.3-2)  
##  compiler      3.4.3   2017-12-01 local          
##  datasets    * 3.4.3   2017-12-01 local          
##  devtools      1.13.3  2017-08-02 CRAN (R 3.2.3) 
##  digest        0.6.13  2017-12-14 cran (@0.6.13) 
##  dplyr       * 0.7.4   2017-09-28 cran (@0.7.4)  
##  evaluate      0.10.1  2017-06-24 cran (@0.10.1) 
##  gdata         2.18.0  2017-06-06 CRAN (R 3.4.2) 
##  ggplot2     * 2.2.1   2016-12-30 cran (@2.2.1)  
##  glue          1.1.1   2017-06-21 cran (@1.1.1)  
##  graphics    * 3.4.3   2017-12-01 local          
##  grDevices   * 3.4.3   2017-12-01 local          
##  grid          3.4.3   2017-12-01 local          
##  gtable        0.2.0   2016-02-26 cran (@0.2.0)  
##  gtools        3.5.0   2015-05-29 CRAN (R 3.4.2) 
##  highr         0.6     2016-05-09 cran (@0.6)    
##  hms           0.3     2016-11-22 cran (@0.3)    
##  htmltools     0.3.6   2017-04-28 cran (@0.3.6)  
##  httr          1.3.1   2017-08-20 CRAN (R 3.4.3) 
##  kableExtra  * 0.6.1   2017-11-01 CRAN (R 3.4.2) 
##  knitr       * 1.17    2017-08-10 cran (@1.17)   
##  labeling      0.3     2014-08-23 cran (@0.3)    
##  lattice       0.20-35 2017-03-25 CRAN (R 3.3.3) 
##  lazyeval      0.2.1   2017-10-29 cran (@0.2.1)  
##  lme4        * 1.1-15  2017-12-21 cran (@1.1-15) 
##  magrittr    * 1.5     2014-11-22 cran (@1.5)    
##  MASS        * 7.3-48  2017-12-24 CRAN (R 3.4.3) 
##  Matrix      * 1.2-11  2017-08-16 CRAN (R 3.4.1) 
##  memoise       1.1.0   2017-04-21 CRAN (R 3.2.3) 
##  methods     * 3.4.3   2017-12-01 local          
##  metR        * 0.1.0   2018-01-09 local          
##  minqa         1.2.4   2014-10-09 cran (@1.2.4)  
##  munsell       0.4.3   2016-02-13 cran (@0.4.3)  
##  nlme        * 3.1-131 2017-02-06 CRAN (R 3.4.0) 
##  nloptr        1.0.4   2017-08-22 cran (@1.0.4)  
##  pkgconfig     2.0.1   2017-03-21 cran (@2.0.1)  
##  plyr          1.8.4   2016-06-08 cran (@1.8.4)  
##  purrr         0.2.4   2017-10-18 cran (@0.2.4)  
##  R6            2.2.2   2017-06-17 CRAN (R 3.2.3) 
##  Rcpp          0.12.13 2017-09-28 cran (@0.12.13)
##  readr         1.1.1   2017-05-16 cran (@1.1.1)  
##  rlang         0.1.4   2017-11-05 cran (@0.1.4)  
##  rmarkdown     1.8     2017-11-17 cran (@1.8)    
##  rprojroot     1.2     2017-01-16 cran (@1.2)    
##  rvest         0.3.2   2016-06-17 CRAN (R 3.4.2) 
##  scales        0.5.0   2017-08-24 cran (@0.5.0)  
##  splines       3.4.3   2017-12-01 local          
##  stats       * 3.4.3   2017-12-01 local          
##  stringi       1.1.5   2017-04-07 cran (@1.1.5)  
##  stringr       1.2.0   2017-02-18 cran (@1.2.0)  
##  tibble        1.3.4   2017-08-22 cran (@1.3.4)  
##  tidyr       * 0.7.2   2017-10-16 cran (@0.7.2)  
##  tidyselect    0.2.2   2017-10-10 cran (@0.2.2)  
##  tools         3.4.3   2017-12-01 local          
##  utils       * 3.4.3   2017-12-01 local          
##  viridisLite   0.2.0   2017-03-24 cran (@0.2.0)  
##  withr         2.1.0   2017-11-01 cran (@2.1.0)  
##  xml2          1.1.1   2017-01-24 CRAN (R 3.4.2) 
##  yaml          2.1.15  2017-12-01 cran (@2.1.15)</code></pre>
=======
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       Europe/Warsaw               
##  date     2018-01-09</code></pre>
<pre><code>## Packages -----------------------------------------------------------------</code></pre>
<pre><code>##  package    * version date       source                        
##  assertthat   0.2.0   2017-04-11 CRAN (R 3.4.0)                
##  backports    1.1.1   2017-09-25 CRAN (R 3.4.2)                
##  base       * 3.4.3   2017-12-07 local                         
##  bindr        0.1     2016-11-13 CRAN (R 3.4.0)                
##  bindrcpp   * 0.2     2017-06-17 CRAN (R 3.4.0)                
##  colorspace   1.3-2   2016-12-14 CRAN (R 3.4.0)                
##  compiler     3.4.3   2017-12-07 local                         
##  datasets   * 3.4.3   2017-12-07 local                         
##  devtools     1.13.3  2017-08-02 CRAN (R 3.4.1)                
##  digest       0.6.13  2017-12-14 cran (@0.6.13)                
##  dplyr      * 0.7.4   2017-09-28 CRAN (R 3.4.2)                
##  evaluate     0.10.1  2017-06-24 CRAN (R 3.4.1)                
##  gdata        2.17.0  2015-07-04 CRAN (R 3.2.0)                
##  ggplot2    * 2.2.1   2016-12-30 CRAN (R 3.4.0)                
##  glue         1.1.1   2017-06-21 CRAN (R 3.4.1)                
##  graphics   * 3.4.3   2017-12-07 local                         
##  grDevices  * 3.4.3   2017-12-07 local                         
##  grid         3.4.3   2017-12-07 local                         
##  gtable       0.2.0   2016-02-26 CRAN (R 3.2.3)                
##  gtools       3.5.0   2015-05-29 CRAN (R 3.2.0)                
##  highr        0.6     2016-05-09 CRAN (R 3.4.0)                
##  hms          0.4.0   2017-11-23 cran (@0.4.0)                 
##  htmltools    0.3.6   2017-04-28 CRAN (R 3.4.0)                
##  httr         1.3.1   2017-08-20 CRAN (R 3.4.1)                
##  kableExtra * 0.5.2   2017-09-15 cran (@0.5.2)                 
##  knitr      * 1.18    2017-12-27 cran (@1.18)                  
##  labeling     0.3     2014-08-23 CRAN (R 3.2.2)                
##  lattice      0.20-35 2017-03-25 CRAN (R 3.4.3)                
##  lazyeval     0.2.0   2016-06-12 CRAN (R 3.4.0)                
##  lme4       * 1.1-15  2017-12-21 cran (@1.1-15)                
##  magrittr   * 1.5     2014-11-22 CRAN (R 3.2.2)                
##  MASS       * 7.3-47  2017-02-26 CRAN (R 3.4.3)                
##  Matrix     * 1.2-12  2017-11-15 CRAN (R 3.4.2)                
##  memoise      1.1.0   2017-04-21 CRAN (R 3.4.0)                
##  methods    * 3.4.3   2017-12-07 local                         
##  metR       * 0.1.0   2018-01-09 local (geneticsMiNIng/metR@NA)
##  minqa        1.2.4   2014-10-09 CRAN (R 3.1.1)                
##  munsell      0.4.3   2016-02-13 CRAN (R 3.2.3)                
##  nlme       * 3.1-131 2017-02-06 CRAN (R 3.4.3)                
##  nloptr       1.0.4   2014-08-04 CRAN (R 3.2.2)                
##  pkgconfig    2.0.1   2017-03-21 CRAN (R 3.4.0)                
##  plyr         1.8.4   2016-06-08 CRAN (R 3.4.0)                
##  purrr        0.2.4   2017-10-18 cran (@0.2.4)                 
##  R6           2.2.2   2017-06-17 CRAN (R 3.4.0)                
##  Rcpp         0.12.14 2017-11-23 cran (@0.12.14)               
##  readr        1.1.1   2017-05-16 CRAN (R 3.4.0)                
##  rlang        0.1.6   2017-12-21 CRAN (R 3.4.3)                
##  rmarkdown    1.8     2017-11-17 cran (@1.8)                   
##  rprojroot    1.2     2017-01-16 CRAN (R 3.4.0)                
##  rvest        0.3.2   2016-06-17 CRAN (R 3.4.0)                
##  scales       0.5.0   2017-08-24 CRAN (R 3.4.1)                
##  splines      3.4.3   2017-12-07 local                         
##  stats      * 3.4.3   2017-12-07 local                         
##  stringi      1.1.6   2017-11-17 cran (@1.1.6)                 
##  stringr      1.2.0   2017-02-18 CRAN (R 3.4.0)                
##  tibble       1.3.4   2017-08-22 CRAN (R 3.4.1)                
##  tidyr      * 0.7.2   2017-10-16 cran (@0.7.2)                 
##  tidyselect   0.2.2   2017-10-10 cran (@0.2.2)                 
##  tools        3.4.3   2017-12-07 local                         
##  utils      * 3.4.3   2017-12-07 local                         
##  withr        2.1.0   2017-11-01 cran (@2.1.0)                 
##  xml2         1.1.1   2017-01-24 CRAN (R 3.4.0)                
##  yaml         2.1.16  2017-12-12 cran (@2.1.16)</code></pre>
>>>>>>> eaba4c4847b664aaeed158990fb581ecda4dedda
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#the-data">The data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#preprocessing">1. Preprocessing</a></li>
      </ul>
</li>
      <li><a href="#tiles-regions">Tiles / regions</a></li>
      <li><a href="#basic-statistics">Basic statistics</a></li>
      <li><a href="#finding-dmr">Finding DMR</a></li>
      <li><a href="#plotting-methylation">Plotting methylation</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aleksandra Brodecka, Przemyslaw Biecek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
